{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "source": [
    "# Task Desription\n",
    "\n",
    "Please read first the following abstract\n",
    "\n",
    "## Abstract\n",
    "                \n",
    "RNA-Seq reveals an unprecedented complexity of the neuroblastoma transcriptome and is suitable for clinical endpoint prediction [ microarray ]\n",
    "\n",
    "### Experiment Description  \n",
    "\n",
    "We generated gene expression profiles from 498 primary neuroblastomas using RNA-Seq and microarrays. We sought to systematically evaluate the capability of RNA deep-sequencing (RNA-Seq)-based classification for clinical endpoint prediction in comparison to microarray-based ones. The neuroblastoma cohort was randomly divided into training and validation sets (**Please note:** <em>in the following we refer to this validation set as test set</em>), and 360 predictive models on six clinical endpoints were generated and evaluated. While prediction performances did not differ considerably between the two technical platforms, the RNA-Seq data processing pipelines, or feature levels (i.e., gene, transcript, and exon junction levels), RNA-Seq models based on the AceView database performed best on most endpoints. Collectively, our study reveals an unprecedented complexity of the neuroblastoma transcriptome, and provides guidelines for the development of gene expression-based predictive classifiers using high-throughput technologies.  Sample clinical characteristics definitions:  \n",
    "\n",
    "* sex: \n",
    "    <ul>\n",
    "    <li>M = male</li>\n",
    "    <li>F = female</li>\n",
    "    </ul>\n",
    "    \n",
    "* age at diagnosis: The age in days at diagnosis \n",
    "    <ul>\n",
    "    <li>integer</li>\n",
    "    </ul>\n",
    "\n",
    "* high risk: Clinically considered as high-risk neuroblastoma\n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "* INSS stage: Disease stage according to International Neuroblastoma Staging System ([INSS](https://www.cancer.org/cancer/neuroblastoma/detection-diagnosis-staging/staging.html)) \n",
    "    <ul>\n",
    "    <li>1</li>\n",
    "    <li>2</li>\n",
    "    <li>3</li>\n",
    "    <li>4</li>\n",
    "    <li>4S</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "* progression: Occurrence of a tumor progression event\n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "\n",
    "* death from disease: Occurrence of death from the disease (yes=1; no=0) \n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Gene expression of 498 neuroblastoma samples was quantified by RNA sequencing as well as by microarray analyses in order to understand the neuroblastoma transcriptome and predict clinical endpoints. \n",
    "\n",
    "\n",
    "## Task\n",
    "\n",
    "The task is to predict the missing values in the validation set (from here on called test set). Do this either with the RNASeq or the Microarray data, or potentially both?\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Code\n",
    "\n",
    "To make your start a bit easier, here is a small notebook reading the data in. It finishes with a function enabling you to save your predictions for submission. \n",
    "\n",
    "#### from here, the code starts\n",
    "\n",
    "First some imports "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt # plotting and visulisation\n",
    "import seaborn as sns # nicer (easier) visualisation\n",
    "%matplotlib inline\n",
    "\n",
    "\n",
    "# for saving\n",
    "import os,os.path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#data_dir ='/rds/homes/b/bjt186/MLP'.format(os.path.sep)\n",
    "\n",
    "#fn_fpkm             = 'log2FPKM.tsv'\n",
    "#fn_patient_info     = 'patientInfo.tsv'\n",
    "#fn_prop_intensities = 'allProbIntensities.tsv'\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the RNAs-Seq data\n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ID</th>\n",
       "      <th>NB001</th>\n",
       "      <th>NB002</th>\n",
       "      <th>NB003</th>\n",
       "      <th>NB004</th>\n",
       "      <th>NB005</th>\n",
       "      <th>NB006</th>\n",
       "      <th>NB007</th>\n",
       "      <th>NB008</th>\n",
       "      <th>NB009</th>\n",
       "      <th>NB010</th>\n",
       "      <th>...</th>\n",
       "      <th>NB489</th>\n",
       "      <th>NB490</th>\n",
       "      <th>NB491</th>\n",
       "      <th>NB492</th>\n",
       "      <th>NB493</th>\n",
       "      <th>NB494</th>\n",
       "      <th>NB495</th>\n",
       "      <th>NB496</th>\n",
       "      <th>NB497</th>\n",
       "      <th>NB498</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <td>0.834381</td>\n",
       "      <td>0.743094</td>\n",
       "      <td>0.909414</td>\n",
       "      <td>0.795775</td>\n",
       "      <td>0.905540</td>\n",
       "      <td>0.869154</td>\n",
       "      <td>1.811352</td>\n",
       "      <td>0.599240</td>\n",
       "      <td>0.981855</td>\n",
       "      <td>1.066399</td>\n",
       "      <td>...</td>\n",
       "      <td>0.997977</td>\n",
       "      <td>1.003559</td>\n",
       "      <td>0.842437</td>\n",
       "      <td>1.057873</td>\n",
       "      <td>0.805515</td>\n",
       "      <td>0.491331</td>\n",
       "      <td>0.868249</td>\n",
       "      <td>0.911379</td>\n",
       "      <td>0.660139</td>\n",
       "      <td>1.152988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.910053</td>\n",
       "      <td>0.941996</td>\n",
       "      <td>1.950857</td>\n",
       "      <td>1.989477</td>\n",
       "      <td>1.942946</td>\n",
       "      <td>1.927608</td>\n",
       "      <td>1.617745</td>\n",
       "      <td>2.161291</td>\n",
       "      <td>1.436439</td>\n",
       "      <td>2.159797</td>\n",
       "      <td>...</td>\n",
       "      <td>2.336929</td>\n",
       "      <td>2.836360</td>\n",
       "      <td>1.205317</td>\n",
       "      <td>2.439868</td>\n",
       "      <td>1.649027</td>\n",
       "      <td>1.451425</td>\n",
       "      <td>1.493852</td>\n",
       "      <td>1.641241</td>\n",
       "      <td>1.994978</td>\n",
       "      <td>1.289534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>1.453191</td>\n",
       "      <td>0.640614</td>\n",
       "      <td>1.156765</td>\n",
       "      <td>1.525277</td>\n",
       "      <td>1.365043</td>\n",
       "      <td>0.899212</td>\n",
       "      <td>1.304178</td>\n",
       "      <td>1.189205</td>\n",
       "      <td>0.771248</td>\n",
       "      <td>1.114787</td>\n",
       "      <td>...</td>\n",
       "      <td>1.182908</td>\n",
       "      <td>1.367371</td>\n",
       "      <td>0.643751</td>\n",
       "      <td>1.096815</td>\n",
       "      <td>0.925425</td>\n",
       "      <td>0.933275</td>\n",
       "      <td>1.208723</td>\n",
       "      <td>0.904511</td>\n",
       "      <td>1.529221</td>\n",
       "      <td>1.102866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.005102</td>\n",
       "      <td>0.005902</td>\n",
       "      <td>0.005192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025347</td>\n",
       "      <td>0.005682</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021880</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024298</td>\n",
       "      <td>0.007295</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006678</td>\n",
       "      <td>0.005746</td>\n",
       "      <td>0.004998</td>\n",
       "      <td>0.004853</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.022780</td>\n",
       "      <td>0.018720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0.580151</td>\n",
       "      <td>0.738233</td>\n",
       "      <td>0.927667</td>\n",
       "      <td>0.936497</td>\n",
       "      <td>0.924853</td>\n",
       "      <td>0.739038</td>\n",
       "      <td>1.018705</td>\n",
       "      <td>0.546324</td>\n",
       "      <td>0.666877</td>\n",
       "      <td>0.865850</td>\n",
       "      <td>...</td>\n",
       "      <td>0.673627</td>\n",
       "      <td>1.401265</td>\n",
       "      <td>0.837443</td>\n",
       "      <td>0.939849</td>\n",
       "      <td>0.743496</td>\n",
       "      <td>0.957837</td>\n",
       "      <td>0.812093</td>\n",
       "      <td>0.488748</td>\n",
       "      <td>1.068072</td>\n",
       "      <td>0.782887</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 498 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ID              NB001     NB002     NB003     NB004     NB005     NB006  \\\n",
       "gene_id                                                                   \n",
       "1/2-SBSRNA4  0.834381  0.743094  0.909414  0.795775  0.905540  0.869154   \n",
       "A1BG         1.910053  0.941996  1.950857  1.989477  1.942946  1.927608   \n",
       "A1BG-AS1     1.453191  0.640614  1.156765  1.525277  1.365043  0.899212   \n",
       "A1CF         0.005102  0.005902  0.005192  0.000000  0.025347  0.005682   \n",
       "A2LD1        0.580151  0.738233  0.927667  0.936497  0.924853  0.739038   \n",
       "\n",
       "ID              NB007     NB008     NB009     NB010  ...     NB489     NB490  \\\n",
       "gene_id                                              ...                       \n",
       "1/2-SBSRNA4  1.811352  0.599240  0.981855  1.066399  ...  0.997977  1.003559   \n",
       "A1BG         1.617745  2.161291  1.436439  2.159797  ...  2.336929  2.836360   \n",
       "A1BG-AS1     1.304178  1.189205  0.771248  1.114787  ...  1.182908  1.367371   \n",
       "A1CF         0.000000  0.000000  0.021880  0.000000  ...  0.024298  0.007295   \n",
       "A2LD1        1.018705  0.546324  0.666877  0.865850  ...  0.673627  1.401265   \n",
       "\n",
       "ID              NB491     NB492     NB493     NB494     NB495     NB496  \\\n",
       "gene_id                                                                   \n",
       "1/2-SBSRNA4  0.842437  1.057873  0.805515  0.491331  0.868249  0.911379   \n",
       "A1BG         1.205317  2.439868  1.649027  1.451425  1.493852  1.641241   \n",
       "A1BG-AS1     0.643751  1.096815  0.925425  0.933275  1.208723  0.904511   \n",
       "A1CF         0.000000  0.006678  0.005746  0.004998  0.004853  0.000000   \n",
       "A2LD1        0.837443  0.939849  0.743496  0.957837  0.812093  0.488748   \n",
       "\n",
       "ID              NB497     NB498  \n",
       "gene_id                          \n",
       "1/2-SBSRNA4  0.660139  1.152988  \n",
       "A1BG         1.994978  1.289534  \n",
       "A1BG-AS1     1.529221  1.102866  \n",
       "A1CF         0.022780  0.018720  \n",
       "A2LD1        1.068072  0.782887  \n",
       "\n",
       "[5 rows x 498 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_fpkm = pd.read_csv('{}/{}'.format(data_dir,fn_fpkm),sep='\\t').rename({'00gene_id':'gene_id'},axis=1)\n",
    "\n",
    "#df_fpkm = pd.read_csv('log2FPKM.tsv.gz',sep='\\t').rename({'00gene_id':'gene_id'},axis=1)\n",
    "\n",
    "#df = pd.read_csv('filename.zip', compression='zip', header=0, sep=',', quotechar='\"')\n",
    "\n",
    "df_fpkm = df_fpkm.set_index(['gene_id'])\n",
    "df_fpkm.columns.name = 'ID'\n",
    "\n",
    "df_fpkm.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the Microarray data\n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_prop_intensities = pd.read_csv('{}/{}'.format(data_dir,fn_prop_intensities),sep='\\t').set_index(['Reporter.Identifier'])\n",
    "#df_prop_intensities.columns.name = 'ID'\n",
    "\n",
    "#df_prop_intensities.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the patient factors, including the potential endpoints \n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. \n",
    "Please note, that the ```FactorValues``` should have a 1-to-1 correspondence to the factors desc ribed in the abstract. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NB498</th>\n",
       "      <td>female</td>\n",
       "      <td>530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB497</th>\n",
       "      <td>female</td>\n",
       "      <td>379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB496</th>\n",
       "      <td>male</td>\n",
       "      <td>132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB495</th>\n",
       "      <td>male</td>\n",
       "      <td>163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB494</th>\n",
       "      <td>male</td>\n",
       "      <td>56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "ID                                                               \n",
       "NB498                   female                             530   \n",
       "NB497                   female                             379   \n",
       "NB496                     male                             132   \n",
       "NB495                     male                             163   \n",
       "NB494                     male                              56   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "ID                                                                        \n",
       "NB498                                      NaN                      NaN   \n",
       "NB497                                      0.0                      0.0   \n",
       "NB496                                      NaN                      NaN   \n",
       "NB495                                      0.0                      0.0   \n",
       "NB494                                      NaN                      NaN   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "ID                                                                \n",
       "NB498                             NaN                        NaN  \n",
       "NB497                               1                        0.0  \n",
       "NB496                             NaN                        NaN  \n",
       "NB495                               1                        0.0  \n",
       "NB494                             NaN                        NaN  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info = pd.read_csv('{}/{}'.format(data_dir,fn_patient_info),sep='\\t').set_index('ID')\n",
    "#df_patient_info = pd.read_csv('patientInfo.tsv',sep='\\t')\n",
    "df_patient_info.columns.name = 'FactorValues'\n",
    "\n",
    "df_patient_info.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####  Divide into training and external testing\n",
    "\n",
    "As you might have already noticed, we removed some of the factor values for some of the patient **ID**s.\n",
    "Every row, where this information is missing indicate a real validation entry. We can use this information and create two separate DataFrames, one for training, one for the validation (testing). \n",
    "\n",
    "The task is to predict the missing values, either with the RNASeq or the Microarray data, or potentially both?\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_patient_info_train  = df_patient_info[df_patient_info['FactorValue..death.from.disease.'].notna()]\n",
    "df_patient_info_test   = df_patient_info[df_patient_info['FactorValue..death.from.disease.'].isna()]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NB497</th>\n",
       "      <td>female</td>\n",
       "      <td>379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB495</th>\n",
       "      <td>male</td>\n",
       "      <td>163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB493</th>\n",
       "      <td>male</td>\n",
       "      <td>190</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB491</th>\n",
       "      <td>male</td>\n",
       "      <td>2326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB489</th>\n",
       "      <td>female</td>\n",
       "      <td>865</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "ID                                                               \n",
       "NB497                   female                             379   \n",
       "NB495                     male                             163   \n",
       "NB493                     male                             190   \n",
       "NB491                     male                            2326   \n",
       "NB489                   female                             865   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "ID                                                                        \n",
       "NB497                                      0.0                      0.0   \n",
       "NB495                                      0.0                      0.0   \n",
       "NB493                                      0.0                      0.0   \n",
       "NB491                                      0.0                      1.0   \n",
       "NB489                                      0.0                      1.0   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "ID                                                                \n",
       "NB497                               1                        0.0  \n",
       "NB495                               1                        0.0  \n",
       "NB493                               1                        0.0  \n",
       "NB491                               4                        1.0  \n",
       "NB489                               4                        0.0  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info_train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NB498</th>\n",
       "      <td>female</td>\n",
       "      <td>530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB496</th>\n",
       "      <td>male</td>\n",
       "      <td>132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB494</th>\n",
       "      <td>male</td>\n",
       "      <td>56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB492</th>\n",
       "      <td>male</td>\n",
       "      <td>947</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB490</th>\n",
       "      <td>female</td>\n",
       "      <td>1759</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "ID                                                               \n",
       "NB498                   female                             530   \n",
       "NB496                     male                             132   \n",
       "NB494                     male                              56   \n",
       "NB492                     male                             947   \n",
       "NB490                   female                            1759   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "ID                                                                        \n",
       "NB498                                      NaN                      NaN   \n",
       "NB496                                      NaN                      NaN   \n",
       "NB494                                      NaN                      NaN   \n",
       "NB492                                      NaN                      NaN   \n",
       "NB490                                      NaN                      NaN   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "ID                                                                \n",
       "NB498                             NaN                        NaN  \n",
       "NB496                             NaN                        NaN  \n",
       "NB494                             NaN                        NaN  \n",
       "NB492                             NaN                        NaN  \n",
       "NB490                             NaN                        NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info_test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(498, 23147)\n",
      "gene_id  1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M  \\\n",
      "NB001       0.834381  1.910053  1.453191  0.005102  0.580151  5.137167   \n",
      "NB002       0.743094  0.941996  0.640614  0.005902  0.738233  5.116157   \n",
      "NB003       0.909414  1.950857  1.156765  0.005192  0.927667  5.405414   \n",
      "NB004       0.795775  1.989477  1.525277  0.000000  0.936497  6.034460   \n",
      "NB005       0.905540  1.942946  1.365043  0.025347  0.924853  5.613060   \n",
      "\n",
      "gene_id     A2ML1     A2MP1    A4GALT     A4GNT  ...     ZWINT      ZXDA  \\\n",
      "NB001    0.027558  0.170671  1.535670  0.057101  ...  5.045111  1.575337   \n",
      "NB002    0.000000  0.000000  1.038092  0.066125  ...  5.037365  1.487028   \n",
      "NB003    0.028036  0.045284  0.943194  0.000000  ...  4.721531  1.508530   \n",
      "NB004    0.000000  0.044204  1.091341  0.028540  ...  4.157084  1.351250   \n",
      "NB005    0.000000  0.000000  1.549078  0.032056  ...  4.766362  1.038738   \n",
      "\n",
      "gene_id      ZXDB      ZXDC    ZYG11A    ZYG11B       ZYX     ZZEF1      ZZZ3  \\\n",
      "NB001    2.395551  3.466692  0.862512  3.561729  5.039327  3.245179  3.296496   \n",
      "NB002    2.748315  3.201112  0.051655  3.595790  5.144157  2.843600  3.052771   \n",
      "NB003    2.482464  3.056684  0.022912  2.236695  4.982975  2.880531  2.431521   \n",
      "NB004    2.447746  3.081720  0.108028  3.312516  6.142091  2.864198  3.119886   \n",
      "NB005    1.890104  2.823576  0.049621  2.235632  5.236531  2.761698  2.226502   \n",
      "\n",
      "gene_id     ID  \n",
      "NB001    NB001  \n",
      "NB002    NB002  \n",
      "NB003    NB003  \n",
      "NB004    NB004  \n",
      "NB005    NB005  \n",
      "\n",
      "[5 rows x 23147 columns]\n"
     ]
    }
   ],
   "source": [
    "x_posed_SEQ = df_fpkm.T\n",
    "#print(x_posed_SEQ.iloc[150:200,:])\n",
    "# load the index as a column called Idents\n",
    "x_posed_SEQ['Idents'] = x_posed_SEQ.index\n",
    "\n",
    "# create Idents as list\n",
    "Idents_list = list(x_posed_SEQ['Idents'])\n",
    "#print(Idents_list[0:100])\n",
    "\n",
    "#str = \"NB013_BC02FPACXX_L1_CGATGT\"\n",
    "#print(str[0:5])\n",
    "\n",
    "#create empty list to hold clean idents\n",
    "trimmed_idents = []\n",
    "\n",
    "# iterate\n",
    "for ident in Idents_list:\n",
    "    new_ident = ident[0:5]\n",
    "    trimmed_idents.append(new_ident)\n",
    "\n",
    "\n",
    "#load clean idents back into dataframe as index\n",
    "x_posed_SEQ.index = trimmed_idents\n",
    "# and as an ID column to allow for the merge\n",
    "x_posed_SEQ['ID'] = trimmed_idents\n",
    "# drop the Idents column\n",
    "clean_SEQ = x_posed_SEQ.drop('Idents', 1)\n",
    "\n",
    "print(clean_SEQ.shape)\n",
    "print(clean_SEQ.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<bound method NDFrame.head of      1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M     A2ML1  \\\n",
      "0       0.834381  1.910053  1.453191  0.005102  0.580151  5.137167  0.027558   \n",
      "1       0.909414  1.950857  1.156765  0.005192  0.927667  5.405414  0.028036   \n",
      "2       0.905540  1.942946  1.365043  0.025347  0.924853  5.613060  0.000000   \n",
      "3       1.811352  1.617745  1.304178  0.000000  1.018705  6.956301  0.000000   \n",
      "4       0.981855  1.436439  0.771248  0.021880  0.666877  5.170205  0.000000   \n",
      "..           ...       ...       ...       ...       ...       ...       ...   \n",
      "244     0.997977  2.336929  1.182908  0.024298  0.673627  5.266052  0.000000   \n",
      "245     0.842437  1.205317  0.643751  0.000000  0.837443  5.049343  0.000000   \n",
      "246     0.805515  1.649027  0.925425  0.005746  0.743496  5.223052  0.000000   \n",
      "247     0.868249  1.493852  1.208723  0.004853  0.812093  6.655558  0.000000   \n",
      "248     0.660139  1.994978  1.529221  0.022780  1.068072  6.581892  0.000000   \n",
      "\n",
      "        A2MP1    A4GALT     A4GNT  ...       ZYX     ZZEF1      ZZZ3     ID  \\\n",
      "0    0.170671  1.535670  0.057101  ...  5.039327  3.245179  3.296496  NB001   \n",
      "1    0.045284  0.943194  0.000000  ...  4.982975  2.880531  2.431521  NB003   \n",
      "2    0.000000  1.549078  0.032056  ...  5.236531  2.761698  2.226502  NB005   \n",
      "3    0.210127  1.959344  0.164813  ...  5.645258  2.755304  2.933508  NB007   \n",
      "4    0.000000  1.374772  0.031028  ...  5.393207  2.730553  2.271136  NB009   \n",
      "..        ...       ...       ...  ...       ...       ...       ...    ...   \n",
      "244  0.053645  0.854657  0.034626  ...  5.103993  3.519328  2.563558  NB489   \n",
      "245  0.000000  1.025780  0.036918  ...  4.938587  2.949716  2.957478  NB491   \n",
      "246  0.050377  0.830303  0.000000  ...  4.877509  2.951646  3.293082  NB493   \n",
      "247  0.124005  1.585030  0.157478  ...  5.207842  3.231953  3.206993  NB495   \n",
      "248  0.000000  1.544703  0.057179  ...  5.427338  3.107112  3.446606  NB497   \n",
      "\n",
      "     FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
      "0                 male                             987   \n",
      "1               female                             625   \n",
      "2               female                             536   \n",
      "3                 male                             942   \n",
      "4               female                             217   \n",
      "..                 ...                             ...   \n",
      "244             female                             865   \n",
      "245               male                            2326   \n",
      "246               male                             190   \n",
      "247               male                             163   \n",
      "248             female                             379   \n",
      "\n",
      "     FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
      "0                                 1.0                      1.0   \n",
      "1                                 1.0                      1.0   \n",
      "2                                 1.0                      1.0   \n",
      "3                                 0.0                      1.0   \n",
      "4                                 0.0                      1.0   \n",
      "..                                ...                      ...   \n",
      "244                               0.0                      1.0   \n",
      "245                               0.0                      1.0   \n",
      "246                               0.0                      0.0   \n",
      "247                               0.0                      0.0   \n",
      "248                               0.0                      0.0   \n",
      "\n",
      "     FactorValue..inss.stage.  FactorValue..progression.  \n",
      "0                           4                        1.0  \n",
      "1                           4                        1.0  \n",
      "2                           4                        1.0  \n",
      "3                           4                        1.0  \n",
      "4                          4S                        0.0  \n",
      "..                        ...                        ...  \n",
      "244                         4                        0.0  \n",
      "245                         4                        1.0  \n",
      "246                         1                        0.0  \n",
      "247                         1                        0.0  \n",
      "248                         1                        0.0  \n",
      "\n",
      "[249 rows x 23153 columns]>\n",
      "23146\n",
      "(249, 23146)\n",
      "   1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M     A2ML1  \\\n",
      "0     0.834381  1.910053  1.453191  0.005102  0.580151  5.137167  0.027558   \n",
      "1     0.909414  1.950857  1.156765  0.005192  0.927667  5.405414  0.028036   \n",
      "2     0.905540  1.942946  1.365043  0.025347  0.924853  5.613060  0.000000   \n",
      "3     1.811352  1.617745  1.304178  0.000000  1.018705  6.956301  0.000000   \n",
      "4     0.981855  1.436439  0.771248  0.021880  0.666877  5.170205  0.000000   \n",
      "\n",
      "      A2MP1    A4GALT     A4GNT  ...    ZWILCH     ZWINT      ZXDA      ZXDB  \\\n",
      "0  0.170671  1.535670  0.057101  ...  3.651545  5.045111  1.575337  2.395551   \n",
      "1  0.045284  0.943194  0.000000  ...  3.547080  4.721531  1.508530  2.482464   \n",
      "2  0.000000  1.549078  0.032056  ...  3.582977  4.766362  1.038738  1.890104   \n",
      "3  0.210127  1.959344  0.164813  ...  3.475448  4.009473  1.407451  2.198063   \n",
      "4  0.000000  1.374772  0.031028  ...  3.581122  5.193176  1.196494  2.289598   \n",
      "\n",
      "       ZXDC    ZYG11A    ZYG11B       ZYX     ZZEF1      ZZZ3  \n",
      "0  3.466692  0.862512  3.561729  5.039327  3.245179  3.296496  \n",
      "1  3.056684  0.022912  2.236695  4.982975  2.880531  2.431521  \n",
      "2  2.823576  0.049621  2.235632  5.236531  2.761698  2.226502  \n",
      "3  3.123347  0.044684  3.130989  5.645258  2.755304  2.933508  \n",
      "4  3.409242  0.012183  2.361504  5.393207  2.730553  2.271136  \n",
      "\n",
      "[5 rows x 23146 columns]\n"
     ]
    }
   ],
   "source": [
    "Merged_Data_patient_train=clean_SEQ.merge(df_patient_info_train, on='ID', how='inner')\n",
    "clean_patient_train = Merged_Data_patient_train.iloc[:,:-7]\n",
    "\n",
    "print((Merged_Data_patient_train).head)\n",
    "\n",
    "lists = clean_patient_train.columns.values.tolist()\n",
    "\n",
    "print(len(lists))\n",
    "print(clean_patient_train.shape)\n",
    "#print(clean_patient_train.head())\n",
    "# ok that looks good so we can drop the id column now\n",
    "#clean_patient_train = clean_patient_train.drop('ID', 1)\n",
    "print(clean_patient_train.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M     A2ML1  \\\n",
      "0     0.743094  0.941996  0.640614  0.005902  0.738233  5.116157  0.000000   \n",
      "1     0.795775  1.989477  1.525277  0.000000  0.936497  6.034460  0.000000   \n",
      "2     0.869154  1.927608  0.899212  0.005682  0.739038  5.281676  0.000000   \n",
      "3     0.599240  2.161291  1.189205  0.000000  0.546324  5.221807  0.008817   \n",
      "4     1.066399  2.159797  1.114787  0.000000  0.865850  4.399096  0.008982   \n",
      "\n",
      "      A2MP1    A4GALT     A4GNT  ...    ZWILCH     ZWINT      ZXDA      ZXDB  \\\n",
      "0  0.000000  1.038092  0.066125  ...  3.617510  5.037365  1.487028  2.748315   \n",
      "1  0.044204  1.091341  0.028540  ...  3.699507  4.157084  1.351250  2.447746   \n",
      "2  0.000000  1.184775  0.032086  ...  3.577320  4.726079  1.567141  2.373818   \n",
      "3  0.000000  0.798254  0.054667  ...  3.550113  3.800683  1.618957  2.620216   \n",
      "4  0.000000  0.850427  0.000000  ...  3.507782  4.289820  0.794625  2.559624   \n",
      "\n",
      "       ZXDC    ZYG11A    ZYG11B       ZYX     ZZEF1      ZZZ3  \n",
      "0  3.201112  0.051655  3.595790  5.144157  2.843600  3.052771  \n",
      "1  3.081720  0.108028  3.312516  6.142091  2.864198  3.119886  \n",
      "2  3.190773  0.264942  3.459969  6.108395  3.970366  3.074821  \n",
      "3  3.262666  0.000000  2.359791  5.018567  2.817902  3.395432  \n",
      "4  2.931842  0.021898  3.284610  5.730512  2.880292  3.263155  \n",
      "\n",
      "[5 rows x 23146 columns]\n",
      "(249, 23146)\n"
     ]
    }
   ],
   "source": [
    "# Clean the test dataset\n",
    "Merged_Data_patient_test=clean_SEQ.merge(df_patient_info_test, on='ID', how='inner')\n",
    "#print(Merged_Data_patient_test.head())\n",
    "\n",
    "clean_patient_test = Merged_Data_patient_test.iloc[:,:-7]\n",
    "#lists = clean_patient_test.columns.values.tolist()\n",
    "#print(lists)\n",
    "#print(clean_patient_test.shape)\n",
    "#print(clean_patient_test.head())\n",
    "# ok that looks good so we can drop the id column now\n",
    "#clean_patient_test = clean_patient_test.drop('ID', 1)\n",
    "print(clean_patient_test.head())\n",
    "print(clean_patient_test.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis\n",
    "\n",
    "From here on, you will need to use your skills ...\n",
    "\n",
    "### Note:\n",
    "To be clear, there are **multiple** target features/attributes to predict. Say you want to build a model predicting **death from disease** of a patient, your target variable is ```'FactorValue..death.from.disease.'``` and the corresponding target vector <em>y</em> would be as follows:\n",
    "\n",
    "```python\n",
    "y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "```\n",
    "Taking the other data into account (RNASeq or microarray) as ``` X_train``` (you will have to preprocess and split this information yourselves), you could for example build a random forest model:\n",
    "\n",
    "```python\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "random_f_model_death = random_f_model = RandomForestClassifier() \n",
    "random_f_model_death.fit(X_train,y_train)\n",
    "```\n",
    "and predict ```y_test``` using ```X_test```. \n",
    "\n",
    "Obviously, you want to avoid any overfitting and might want to use appropriate validation approaches. \n",
    "\n",
    "Once you have your model and the prediction for the test data, you should be able to fill the ```'FactorValue..death.from.disease.'``` column in the test set. \n",
    "\n",
    "For the submission, please also include the confidence/probability/score for each of the prediction (assume ```1``` to be the value for the positive class). This only applies to the factors:  **high risk**, **progression** and **death from disease** . \n",
    "\n",
    "The other two factors need to be treated differently.  \n",
    "\n",
    "Using this very rough outline, you should be able to predict all factors in the test data. \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNASeq\n",
      "(23146, 498)\n",
      "Patient info train\n",
      "(249, 6)\n",
      "Patient info test\n",
      "(249, 6)\n"
     ]
    }
   ],
   "source": [
    "print(\"RNASeq\")\n",
    "\n",
    "print(np.shape(df_fpkm))\n",
    "\n",
    "# size of the patient info training set\n",
    "\n",
    "print(\"Patient info train\")\n",
    "\n",
    "print(np.shape(df_patient_info_train))\n",
    "\n",
    "# and the size of the patient info test set\n",
    "\n",
    "print(\"Patient info test\")\n",
    "\n",
    "print(np.shape(df_patient_info_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['FactorValue..Sex.', 'FactorValue..age.at.diagnosis.', 'FactorValue..death.from.disease.', 'FactorValue..high.risk.', 'FactorValue..inss.stage.', 'FactorValue..progression.']\n",
      "['1', '1', '1', '4', '4', '4', '3', '1', '4', '1']\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXAAAAD4CAYAAAD1jb0+AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+j8jraAAALbElEQVR4nO3dUWyd91nH8e+PpFM3yrR0darQNHhIUbeqUltkjUAlLpZNKsu09IJKBVpFU6dwsUGGJk0eNwOucoHGuECTonUQsWqo6ooardJYlLVCSCiQtAHWeVXGCKXMNBmobHCx0fJw4ReWOnZ97GP7+Em+Hyk65319jv+PXul88+Y9Pk6qCklSPz826QEkSWtjwCWpKQMuSU0ZcElqyoBLUlPbN3Oxm266qaanpzdzSUlq7+zZs9+tqqnF+zc14NPT05w5c2Yzl5Sk9pL801L7vYQiSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTW3qJzG1OtOzT01k3QtHD0xkXUmr4xm4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTIwU8yW8meT7J15N8Mcn1SW5McjLJ+eF2x0YPK0n6kRUDnuQW4DeAmaq6A9gGPADMAqeqai9watiWJG2SUS+hbAfenGQ78BbgO8BB4Pjw9ePAfes/niRpOSsGvKr+Bfg94EVgHviPqvoqcHNVzQ+PmQd2LvX8JIeTnEly5tKlS+s3uSRd40a5hLKDhbPtdwA/Cfx4kgdHXaCqjlXVTFXNTE1NrX1SSdLrjHIJ5b3AP1bVpar6b+AJ4OeBl5PsAhhuL27cmJKkxUYJ+IvAviRvSRJgPzAHnAAODY85BDy5MSNKkpayfaUHVNXpJI8DzwKvAs8Bx4AbgMeSPMxC5O/fyEElSa+3YsABqupTwKcW7f4BC2fjkqQJ8JOYktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJampkQKe5G1JHk/yzSRzSX4uyY1JTiY5P9zu2OhhJUk/MuoZ+B8AX6mqdwJ3AnPALHCqqvYCp4ZtSdImWTHgSd4K/ALwCEBV/bCqXgEOAseHhx0H7tuoISVJV9o+wmN+GrgE/FGSO4GzwBHg5qqaB6iq+SQ7l3pyksPAYYA9e/asedDp2afW/NxxXTh6YGJrS9JyRrmEsh34GeCzVXU38F+s4nJJVR2rqpmqmpmamlrjmJKkxUYJ+EvAS1V1eth+nIWgv5xkF8Bwe3FjRpQkLWXFgFfVvwL/nOS2Ydd+4BvACeDQsO8Q8OSGTChJWtIo18ABfh14NMmbgG8DH2Ih/o8leRh4Ebh/Y0aUJC1lpIBX1TlgZokv7V/fcSRJo/KTmJLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU2NHPAk25I8l+TLw/aNSU4mOT/c7ti4MSVJi63mDPwIMHfZ9ixwqqr2AqeGbUnSJhkp4El2AweAz122+yBwfLh/HLhvfUeTJL2R7SM+7jPAJ4CfuGzfzVU1D1BV80l2LvXEJIeBwwB79uwZY1RJ62l69qmJrX3h6IGJrX01WfEMPMkHgItVdXYtC1TVsaqaqaqZqamptXwLSdISRjkDvwf4YJL3A9cDb03yBeDlJLuGs+9dwMWNHFSS9HornoFX1SerandVTQMPAF+rqgeBE8Ch4WGHgCc3bEpJ0hXG+Tnwo8D7kpwH3jdsS5I2yahvYgJQVc8Azwz3/w3Yv/4jSZJG4ScxJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpqVX9jzzS1Wp69qmJrX3h6IGJra3ePAOXpKYMuCQ1ZcAlqSmvgUu6Zlxt73V4Bi5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekplYMeJJbkzydZC7J80mODPtvTHIyyfnhdsfGjytJ+j+jnIG/Cny8qt4F7AM+kuR2YBY4VVV7gVPDtiRpk6wY8Kqar6pnh/vfB+aAW4CDwPHhYceB+zZqSEnSlVZ1DTzJNHA3cBq4uarmYSHywM5lnnM4yZkkZy5dujTetJKk/zdywJPcAHwJ+FhVfW/U51XVsaqaqaqZqamptcwoSVrCSAFPch0L8X60qp4Ydr+cZNfw9V3AxY0ZUZK0lFF+CiXAI8BcVX36si+dAA4N9w8BT67/eJKk5Yzyv9LfAzwE/H2Sc8O+3wKOAo8leRh4Ebh/Y0aUJC1lxYBX1V8CWebL+9d3HEnSqPwkpiQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlNjBTzJvUleSPKtJLPrNZQkaWVrDniSbcAfAr8I3A78cpLb12swSdIbG+cM/N3At6rq21X1Q+BPgYPrM5YkaSWpqrU9Mfkl4N6q+vCw/RDws1X10UWPOwwcHjZvA15Y46w3Ad9d43OvRR6v1fF4rY7Ha/XGOWY/VVVTi3duH2OYLLHvir8NquoYcGyMdRYWS85U1cy43+da4fFaHY/X6ni8Vm8jjtk4l1BeAm69bHs38J3xxpEkjWqcgP8NsDfJO5K8CXgAOLE+Y0mSVrLmSyhV9WqSjwJ/DmwDPl9Vz6/bZFca+zLMNcbjtToer9XxeK3euh+zNb+JKUmaLD+JKUlNGXBJamrLBzzJ55NcTPL1Sc/SSZJtSZ5L8uVJz7KVJbk+yV8n+dskzyf5nUnPtJUluTXJ00nmhuN1ZNIzbVWLX4NJ9iU5neTccPx+e9w1tnzAgT8G7p30EA0dAeYmPUQDPwDeU1V3AncB9ybZN+GZtrJXgY9X1buAfcBH/BUay1r8GjwOHK6qu4A7gMfGXWDLB7yq/gL490nP0UmS3cAB4HOTnmWrqwX/OWxeN/zxnf1lVNV8VT073P8+C4G6ZbJTbT3LvAZ3AvMAVfVaVX1j3HW2fMC1Jp8BPgH8z6QH6WD4p+454CJwsqpOT3qmDpJMA3cDHq8rLfUa/H3ghSR/luTXklw/7iIG/CqT5APAxao6O+lZuhjOhu5i4dPE705yx6Rn2uqS3AB8CfhYVX1v0vNsJcu9Bqvqd4EZ4KvArwBfGXctA371uQf4YJILLPyGyPck+cJkR+qhql4BnsH3XN5QkutYiPejVfXEpOfZgpZ9DVbVP1TVZ4H9wJ1J3j7OQgb8KlNVn6yq3VU1zcKvN/haVT044bG2rCRTSd423H8z8F7gm5OdautKEuARYK6qPj3pebai5V6DSQ4Mxw9gL/Aa8Mo4a235gCf5IvBXwG1JXkry8KRn0lVlF/B0kr9j4ff7nKwqf/RyefcAD7FwVnlu+PP+SQ/VxEMsXAM/B/wJ8KtV9do439CP0ktSU1v+DFyStDQDLklNGXBJasqAS1JTBlySmjLgktSUAZekpv4XkUl2466yqCoAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# have a look at the INSS classifications\n",
    "\n",
    "headers_list = list(df_patient_info_train.columns)\n",
    "print(headers_list)\n",
    "INSS_list = list(df_patient_info_train[\"FactorValue..inss.stage.\"]) #INSS Stage\n",
    "print(INSS_list[0:10])\n",
    "\n",
    "from collections import Counter\n",
    "INSS_cats = Counter(INSS_list)\n",
    "plt.hist(INSS_list)\n",
    "INSS_dict = {\"1\":60, \"2\":40, \"3\":30, \"4\":91, \"4S\":28}\n",
    "\n",
    "# decision taken to hold off on re-classifying\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RandomForestClassifier()"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "# using the RNA data as the X train\n",
    "\n",
    "#X_train = df_fpkm[0:249]\n",
    "\n",
    "#looking for the people dying from the disease as the training set\n",
    "#y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "\n",
    "#random_f_model_death = random_f_model = RandomForestClassifier()\n",
    "#random_f_model_death.fit(X_train,y_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0      0.834381\n",
      "1      0.909414\n",
      "2      0.905540\n",
      "3      1.811352\n",
      "4      0.981855\n",
      "         ...   \n",
      "244    0.997977\n",
      "245    0.842437\n",
      "246    0.805515\n",
      "247    0.868249\n",
      "248    0.660139\n",
      "Name: 1/2-SBSRNA4, Length: 249, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "#print(Merged_Data_patient_train)\n",
    "first_column = Merged_Data_patient_train.iloc[:, 0]\n",
    "print(first_column)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FactorValues  FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
      "ID                                                                \n",
      "NB497                         0                             379   \n",
      "NB495                         1                             163   \n",
      "NB493                         1                             190   \n",
      "NB491                         1                            2326   \n",
      "NB489                         0                             865   \n",
      "...                         ...                             ...   \n",
      "NB009                         0                             217   \n",
      "NB007                         1                             942   \n",
      "NB005                         0                             536   \n",
      "NB003                         0                             625   \n",
      "NB001                         1                             987   \n",
      "\n",
      "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
      "ID                                                                        \n",
      "NB497                                      0.0                      0.0   \n",
      "NB495                                      0.0                      0.0   \n",
      "NB493                                      0.0                      0.0   \n",
      "NB491                                      0.0                      1.0   \n",
      "NB489                                      0.0                      1.0   \n",
      "...                                        ...                      ...   \n",
      "NB009                                      0.0                      1.0   \n",
      "NB007                                      0.0                      1.0   \n",
      "NB005                                      1.0                      1.0   \n",
      "NB003                                      1.0                      1.0   \n",
      "NB001                                      1.0                      1.0   \n",
      "\n",
      "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
      "ID                                                                \n",
      "NB497                               1                        0.0  \n",
      "NB495                               1                        0.0  \n",
      "NB493                               1                        0.0  \n",
      "NB491                               4                        1.0  \n",
      "NB489                               4                        0.0  \n",
      "...                               ...                        ...  \n",
      "NB009                              4S                        0.0  \n",
      "NB007                               4                        1.0  \n",
      "NB005                               4                        1.0  \n",
      "NB003                               4                        1.0  \n",
      "NB001                               4                        1.0  \n",
      "\n",
      "[249 rows x 6 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/rds/bear-apps/2020a/EL8-cas/software/SciPy-bundle/2020.03-foss-2020a-Python-3.8.2/lib/python3.8/site-packages/pandas/core/generic.py:6746: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._update_inplace(new_data)\n"
     ]
    }
   ],
   "source": [
    "#df_patient_info_train['FactorValue..Sex._male.'].replace({'male':1, 'female':0}, inplace=True)\n",
    "df_patient_info_train['FactorValue..Sex.'].replace({'male':1, 'female':0}, inplace=True)\n",
    "print(df_patient_info_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>NB497</th>\n",
       "      <td>0</td>\n",
       "      <td>379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB495</th>\n",
       "      <td>1</td>\n",
       "      <td>163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB493</th>\n",
       "      <td>1</td>\n",
       "      <td>190</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB491</th>\n",
       "      <td>1</td>\n",
       "      <td>2326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB489</th>\n",
       "      <td>0</td>\n",
       "      <td>865</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB009</th>\n",
       "      <td>0</td>\n",
       "      <td>217</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4S</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB007</th>\n",
       "      <td>1</td>\n",
       "      <td>942</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB005</th>\n",
       "      <td>0</td>\n",
       "      <td>536</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB003</th>\n",
       "      <td>0</td>\n",
       "      <td>625</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NB001</th>\n",
       "      <td>1</td>\n",
       "      <td>987</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>249 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues  FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "ID                                                                \n",
       "NB497                         0                             379   \n",
       "NB495                         1                             163   \n",
       "NB493                         1                             190   \n",
       "NB491                         1                            2326   \n",
       "NB489                         0                             865   \n",
       "...                         ...                             ...   \n",
       "NB009                         0                             217   \n",
       "NB007                         1                             942   \n",
       "NB005                         0                             536   \n",
       "NB003                         0                             625   \n",
       "NB001                         1                             987   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "ID                                                                        \n",
       "NB497                                      0.0                      0.0   \n",
       "NB495                                      0.0                      0.0   \n",
       "NB493                                      0.0                      0.0   \n",
       "NB491                                      0.0                      1.0   \n",
       "NB489                                      0.0                      1.0   \n",
       "...                                        ...                      ...   \n",
       "NB009                                      0.0                      1.0   \n",
       "NB007                                      0.0                      1.0   \n",
       "NB005                                      1.0                      1.0   \n",
       "NB003                                      1.0                      1.0   \n",
       "NB001                                      1.0                      1.0   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "ID                                                                \n",
       "NB497                               1                        0.0  \n",
       "NB495                               1                        0.0  \n",
       "NB493                               1                        0.0  \n",
       "NB491                               4                        1.0  \n",
       "NB489                               4                        0.0  \n",
       "...                               ...                        ...  \n",
       "NB009                              4S                        0.0  \n",
       "NB007                               4                        1.0  \n",
       "NB005                               4                        1.0  \n",
       "NB003                               4                        1.0  \n",
       "NB001                               4                        1.0  \n",
       "\n",
       "[249 rows x 6 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info_train.rename(columns={'FactorValue..inss.stage._4S' : 'FactorValue..inss.stage._5'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RandomForestClassifier()"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "X_train = clean_patient_train\n",
    "\n",
    "# looking for the people dying from the disease as the training set\n",
    "y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "\n",
    "random_f_model_death = random_f_model = RandomForestClassifier()\n",
    "random_f_model_death.fit(X_train,y_train)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Validation Step"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import train_test_split function\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "X=clean_patient_train  # Features\n",
    "y=df_patient_info_train['FactorValue..death.from.disease.'].astype(int)  # Labels\n",
    "\n",
    "# Split dataset into training set and test set\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "clf=RandomForestClassifier(n_estimators=100)\n",
    "\n",
    "#Train the model using the training sets y_pred=clf.predict(X_test)\n",
    "clf.fit(X_train,y_train)\n",
    "\n",
    "# predictions\n",
    "y_pred=clf.predict(X_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 0.8\n"
     ]
    }
   ],
   "source": [
    "#Import scikit-learn metrics module for accuracy calculation\n",
    "from sklearn import metrics\n",
    "# Model Accuracy, how often is the classifier correct?\n",
    "print(\"Accuracy:\",metrics.accuracy_score(y_test, y_pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RNF103-VPS24    0.004741\n",
       "SCYL3           0.004610\n",
       "CADM1           0.003914\n",
       "AHCYL2          0.003856\n",
       "AMH             0.003634\n",
       "                  ...   \n",
       "PABPC1L2A       0.000000\n",
       "PABPC1L2B       0.000000\n",
       "PABPC1P2        0.000000\n",
       "PABPC3          0.000000\n",
       "1/2-SBSRNA4     0.000000\n",
       "Length: 23146, dtype: float64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "feature_imp = pd.Series(clf.feature_importances_,index=lists).sort_values(ascending=False)\n",
    "feature_imp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No handles with labels found to put in legend.\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAcAAAAEWCAYAAADxQkdBAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+j8jraAAAgAElEQVR4nO3deZxcVZn/8c+XNYSdsAgECLssMgEjiIqGQQXcgFGEgGgcR7YBHB10UFyiiCCjAyIqkxkQRFBQVuXHNmjcAJEl7DtEDSAEEEJYAkm+vz/OqeR2UdVdVV3Vdbvqeb9e/aL63O3cm6ZPn3PPcx7ZJoQQQug3S3W7AiGEEEI3RAMYQgihL0UDGEIIoS9FAxhCCKEvRQMYQgihL0UDGEIIoS9FAxh6hqS7JE3u8DUsabP8+XRJX2rgmHmSNulkvUIIzYsGMIwKkq6S9LUa5XtJ+pukZWxvY3vGSNXJ9qG2j2tgv5VsP9zu60uaJunH7T5vKyRNlfT7Np5vyHuTNEvSS/kPjMrXesO87ixJ7xzOOcLoEQ1gGC3OAg6SpKryg4BzbS8Y+SoFAEnLdPHy789/YFS+HutiXbr9LEKTogEMo8UlwBrALpUCSasD7wN+lL9f/Ne7pB0l3SRprqQnJP1XLp8saXbxxDWOu17Ss5Iel3SapOVqVUjSWZK+nj//oqonskjS1LytOGx6lqTvSbpc0vOS/ihp08I53y3pPknPSfq+pN9I+pdGHlC+zuGSHsjnPk7Spvl+5kq6oHIvlecg6QuSnsrP4MDCuVaV9CNJcyT9WdIXJS2Vt02V9AdJJ0t6BjgfOB3YOd/7s3m/90q6NV/7r5KmFc4/Idf3Y5L+kutwbN62B/AFYL98vtsauf+qup+R//0elfR1SUvnbZtK+pWkp/M1z5W0Wt52DrAhUPm3/FwDPy/TJP1c0o8lzQWmDnH9zfK/6XP5+uc3c2+hvaIBDKOC7ZeAC4CPFoo/DNxru9YvyO8A37G9CrBpPrYRC4FPA2sCOwO7AYc3UL/FPRHgQ8DfgGvr7D4F+CqwOvAgcDyApDWBnwOfB8YB9wFvabDeFXsAbwTeDHwOmA4cCGwAbJuvXfE60n2uD3wMmC5py7ztu8CqwCbAO0jP/eOFY3cCHgbWBj4CHApcn5/BanmfF/JxqwHvBQ6TtHdVfd8GbEl6zl+WtJXtK4FvAOfn8/1Dk8/gbGABsBmwPfBuoPJHhIATgPWArfJzmQZg+yDgLyzpVZ7U4PX2Iv27rQacO8T1jwOuJv3bjyc959Al0QCG0eRsYF9JK+TvP5rLankV2EzSmrbn2b6hkQvYvtn2DbYX2J4F/DepAWiIpC1IPdL9bP+1zm4X2b4xD9ueC0zM5e8B7rJ9Ud52KqkhbcY3bc+1fRdwJ3C17YdtPwdcQfqFXPQl2/Nt/wa4HPhw7q3sB3ze9vP5OXybNNxc8Zjt7+bn9FKtitieYfsO24ts3w78hNc+y6/afin/EXMb0Gxjd0nurT8r6RJJ6wB7Av9m+wXbTwInA/vnOj1o+5p8z3OA/6pRp2Zdb/sS24uAVQa7PunnciNgPdsv227be9PQvGgAw6iRf1nMAfZSmlX5JuC8Ort/AtgCuFfSnyS9r5FrSNpC0i+VJtbMJfVE1mzw2FWBS0mNyu8G2bXYqL0IrJQ/rwcsbjSdVqofMPzWgCcKn1+q8f1Khe//bvuFwvd/znVYE1guf1/ctn7h+3qN+2KSdpL06zyM+hypl1j9LOs9i0btbXu1/LU3qXFZFni80jCS/ohZO9dpbUk/zUOTc4Ef16hTs4rPYtDrk3rlAm5UmrX8z8O8dhiGaADDaPMjUs/vIFLv5olaO9l+wPYU0i+ebwI/l7QiaVhubGW/3NtZq3DoD4B7gc3z8OkXSL+wBpXfj50H/Nr2f7dyY8DjpGGxyjlV/L4DVs/PpGJD4DHgKZb0VIrbHi18X51GplZamfOAy4ANbK9Kek845LMc5HyN+CswH1iz0DCuYnubvP2EfO7t8r/vR6rqVH3doX5eqo8Z9Pq2/2b7k7bXAw4Bvq/8fjiMvGgAw2jzI+CdwCepP/yJpI9IWisPSz2bixcC9wNj8gSNZYEvAssXDl0ZmAvMk/R64LAG63U8sCLwqWZupsrlwBsk7a00m/BfSe/pOumrkpaTtAtpQtHPbC8kvTM9XtLKkjYCPkPqLdXzBDBeAycMrQw8Y/tlSTsCBzRRryeACZWJN42y/TjpHdu3Ja0iaak88aUyzLkyMA94VtL6wGdrXLcYsznUz0tT15e0r6TKHzV/JzWeC5u5x9A+0QCGUSW/j7qO1NhcNsiuewB3SZpHmhCzf37n8hxpUsv/kno0LzBwmPFo0i/q54H/Ic1wbMQU0sSTv2vJTNADhzqoyPZTwL7AScDTwNbATaQeRSf8jfRL+DHSu8hDbd+btx1JejYPA78n9ebOHORcvwLuAv4m6alcdjjwNUnPA1+m8YlIAD/L/31a0i1NHAdphGA54G7S/f0cWDdv+yqwA/Ac6Q+Oi6qOPQH4Yh6+PLqBn5dmr/8m4I/55/Iy4FO2H2ny/kKbKBLihlBOufczGzjQ9q/bfO7JwI9td3KINYRSix5gCCUiaXdJq0laniXvHxuawRpCaE40gICkfZSCcl9fVb59Lt89f3+xpJmSHsyBrDPz11skzVAKYK6Ufag7dxNGuZ2Bh0gTUd5PmuVYM8wghDA8MQQKSLqANEZ/re1phfKTyL+QbE8tlE8Gjrb9vkLZjFx208jUOoQQwnD0fQ9Q0krAW0lxY/sXykVa0WMq8G5JY7pSwRBCCB0RC7fC3sCVtu+X9IykHWzfQmoUH7H9UO7dvYfXzhirdq6kynDVbrafrt5B0sHAwQBjl1/+jZuvM6zF60MovWXWWqPbVQg95uabb37KdnU8ZtOiAUzT10/Jn3+av78l//enhfKDGLoBPHCoIVDb00nrMzJxo018zTGvyfATQk9Z67CPdLsKocdI+vPQew2trxtASeOAfwS2lWRgacCSjgE+CHxAaYV6AeMkrWz7+e7VOIQQQrv0dQNIesf3I9uHVAok/Ya02sNttncvlJ9NGi49p10XX2atNeKv4xBC6JJ+nwRzJHBwVfjDhcC7SBNfdq8q/5Kku4AzgF0k7QSQF1p+I3CepLslHUIIIYRS6/ceYGWpov1ZkhPs1LxWn0nvAa/K+84hLU812fZ8pdxty+X1AacDW9menQOYJzRy8QVznmHODwZbXjGE0S9GOcJgXn31VWbPns3LL7/8mm1jxoxh/PjxLLvssh25dt82gIXwh11Ja/JNy+WV8Id3Ab+TNMb2y6Q4wadsz4fF6zYiaQ3Sc3w6l88nJTINIYQwhNmzZ7PyyiszYcIE0q/fxDZPP/00s2fPZuONN+7Itft5CHRx+APwjKQdcvni8AdgBin8AdIK7xtIul/S9yuru9t+htSA/lnSTyQd2OwK9iGE0K9efvllxo0bN6DxA5DEuHHjavYM26Wff1FXhzlMGazc9jzSe76DScOh50uamrf9C7AbcCMpm0DdVfMlHSzpJkk3PT1vbjvvJ4QQRqXqxm+o8nbpyyHQVsMfcp60GcAMSXcAHwPOArB9B3CHpHOAR0gryLxGdRxgx24yhBDCoPq1B1gJf9jI9gTbG5AarUr4wwa5fCPS7M+9JW0pafPCOSaShj1XymuDDigfofsIIYTQor7sAZKGNU+sKrsQ2BO4uEb5YaQZo9+VtBqwAHiQNBwq4HOS/ht4iZQwc2ojlYg4wBBCSBNeag13djpZQ+kaQEn7kJYc26qQnRpJ25OWKNvD9lWF8mNJGbwXAouAQ2z/Ma/fuS4pm/ZywP8BX7T9rO3JOSPzSoVLzyVlfThd0jTgk6R3fcsAX7B9s6SrC+VbAG8B3gCsR2r83gDcAfyvpDNtnzrYvUYYROgX8YdeqGfMmDE8/fTTr5kIU5kFOmZM5/IQlK4BJPXOfk8hNq+qfHFsnqSdgfcBOxRj8wrHHGj7JknLAScAlwLvaLAeJ9v+lqStSOEQa9cqB9a2fXyuzzzbE5u/5RBC6E/jx49n9uzZzJkz5zXbKnGAnVKqBrBdsXnVbL8i6XPAg5L+wfZtjdbJ9j2SFgBrDlL+ZHN3GkIIAWDZZZftWJzfUMo2CaYtsXm15BmctwGVZc9WKGRvnwnUTMuQlztbRBr2HLJ8KBEGEUII5VC2BrBtsXl1FN+yvmR7YuUL+HLVvp/ODeO3gP285G1svfKG2J5ue5LtSeNWWqWZQ0MIIbRRaYZAOxGbV3X+pUmTVO5psEon2/5WE+UhhBBGkdI0gLSQmkjSjcAi2w/kTTVj8PKC1ccDf7V9ewfvoSkRBhFCCN1TpgawnbF5kCan/EnS/fn7/wP2KoRTVC8w9z5gV0lvB9YGLgCoCqdYD7hX0v/afjZvn2d7cThFHoKdZPuIoW44wiBCP4k/9kLZlKYBtD25RtmpwGti6WxfRpolCikWr5ZKqqNrbU+rFEqqhFM8VCjbGVgfWG844RS2VxriHWQIIYSSKNskmLYohFN8ghRPWCmvhFNMJSW8rURYviacwvZj1ee1/QrwOWBDSf/Q0ZsIIYTQUT3ZAFLCcIqKCIMIIYRy6NUGsEzhFANEGEQIIZRDad4BtksJwylCCCGUUM81gIyicIoIgwghhO7piQawmEGCQjhFIeThB8CbyeEUkhaSsjasTgqz2Bc4IYdTrA0sCzwq6U+kLBPnStoYMDAPmCPp68BxhTpcma8xizTLdEgRBhH6SfyxF8qmJxpAChkkqsIpKuVjbe9ZKH+pkrUh9wLfZvstOUTig8CHbS+SNB54wfbfJc0ixfc9lWeZTgemF2IA/xMYS0rHNGQMYAghhO4a9ZNgWgh5qHY9KQYQUjjE47YXAdiebfvv1QfkSTOHkoZP18hl1wLPt+OeQgghdN6obwBpPuRhsTyhZTeWBNVfALw/hzR8Ow+h1mR7LvAIsHkzlY0wiBBCKIdeaACbCnnIVsgxe08DawDXQOrxAVsCnyelOrpW0m6DXFuDbKspwiBCCKEcRvU7wFZDHsjvACWtCvwS+Ffykmt5NZgrgCskPUHqYV5b49orAxOA+6u3hRBCKL9R3QDSQsgDcE6lzPZzko4CLpX0A1J8399sPyZpKWA74DXhDvm94/eBS2q9I2xUhEGEEEL3jPYh0CnUzhRRDHnYJ/cObwAOqOwkaftcvjZpabP98+dHJb0EvAhMptBgkjJBvAw8RXp3+A1JYyVdLulFUsaJ3SXNlrQ7IYQQSktNJjQfdSRdQJrdWZ0V4iRgZ+Ah21ML5YvTG+Ve4/22j68XIkFKk7ST7V/nbBHXAt+wfcVQdZu40Sa+5phBlw4NoefEqEcYLkk325403POM9h7goEYiRML2i7Z/ncteIQXej+/A7YQQQmijnm4AGeEQibySzPupMWkmhBBCufR6AzhiIRKSlgF+Apxq++F6FYo4wBBCKIeefQeYQyRmA0+S1vBcOv93Y+BR4FXSOp8CxgHr2n6+8g6wECLxs5yZvvr8RwMb2T4yf38mMM/2UY3WMd4Bhn4U7wDDcLXrHeBoD4MYzIiFSOSFsVcF/qWZCkYYRAghdE8vD4EOGSJRVX5AVRm2b2VgiMQvJN1JavgWAKfl2aDHAlsDt+R3hE01hCGEEEZezw6BNqKYRsn2vYXyShqlPWxfVSivpFFahrQO6EG2n809wlNIq9IYeJkULvHIYNePIdDQr2LkIwxHhEG0x+I0SnXKp1SVv2R7ou1tgWdIS6gB7AesB2xn+w3APsCzHat1CCGEYevbBrAbaZRCCCGUR982gHQpjVKEQYQQQjn0cwPYlTRKkQ4phBDKoZfDIOrqZhqlEEII5dCXDSBdSqNULeIAQwihe3q+AawT6jAF+Fnu/VVCHS4E9iRNfPm27X/P+y4EvkyhAZR0G3A3KUbwe6T3gevmCTOvkLJEPDFU3RbMeYY5P/hxG+4yhNEr/ggM3dIP7wBfE+pgezKwEYVQB9un2t6TlN7onyStmXe/Gzg377OSpK1Iz+3twP62D7a9qe2xwF+A8bbH2f7siNxdCCGElvR0A9hiqMMCYDrw6TqnPYDUG7wa+ED7ax1CCGEk9HQDSOuhDt8DDsyTXartB5xPyvxQHSg/pAiDCCGEcuj1BrCVUAdszwV+BAzI7CDpTcAc238mzfDcQdLqzVQowiBCCKEcenYSzDBCHSpOIa0H+sNC2RTg9ZJm5e9Xyef6347eTAghhLbr2QaQ4Yc6PCPpAtL7wzNzeMO+pPU+H83H7ZrP11IDGGEQIYTQPb3cAE4BTqwqq4Q6PJJ7hZXQiAuBw3KqoxUl7Z5DI74NHAF8BXgL8KjtR3P298eBPwJbS/oUqSFcFfiDpG/Z/p+hKhhhECFEGETonp59B2h7su0rq8oqoQ5rUAiNsH1ZLh+QBcL2Ezm84QVgLWDXfKp3kbLKY3td4DngfNvLAbsA35C0TodvMYQQwjD0bANYzzCyQFwBvDd/nkKaBfoatp8EHiLFGYYQQiipvmsAaT004qfA/rlh3I40/PkakjYBNgEerLM9wiBCCKEE+rEBbDU04nZgQi7/fzXOu1/OFPET4BDbz9S6eIRBhBBCOfTyJJjXaENoxGXAt4DJwLiq059v+4hO30MIIYT26KsGkGGGRgBnAs/ZvkPS5OFWJsIgQgihe/ptCHQKcHHlm5wp4u2kbA7F8u2Bj1K1Egxwr+3v5M87AbtK2hDYAviwpAWSPtTB+ocQQmiTvuoB5iwQRZWwh1/ZPr1G+UP5uJWKB+Vs7weTguL/Imk66b3h0c3UJ+IAQ0hiJCR0Q7/1ABdrNRxC0i7A/wDvzTNGsT0rT5JZNDK1DyGEMFx92wDSWjjE8sClwN6F5LohhBBGoX5uAFsJh3gVuI7Ua2xJxAGGEEI59NU7wIphhEMsAj4M/J+kL9j+RrPXtj2dlHCXiRtt4vbcUQghhGb1ZQPIMMIhbL8o6X3A7yQ9YfuMVisRYRAhhNA9/ToEOiAcIrsQeHOd8gOKBXmVlz2AL0raS9KbJM0mpUv6b0l3dabaIYQQ2qWne4A5zu8ilqQ9qvg0cIukPXLaI2yfKukK4GRJnwWeBeYCX8mZIrC9kqRLgbVt7wxsnK8zjTRc+gAwBvitpKVsDzorNMIgQlgiRkPCSOv1HmAlnm//OuWLJ7jkcIfLgem2N7X9RuBI0sLWlX1WA3YAVpO0cdU5T7Y9EdgaeAPwjjbfSwghhDbq2QawhTi/A4HrbV9W2df2nbbPKpz2g8AvyJkh6lx6OVIv8O9tuZEQQggd0bMNIM3H+W0D3DLEOSt5AH9CVbYI4NM5G8TjwP22Z9Y6QYRBhBBCOfRyA9hS2qMKSRdLulPSRfn7dYDNgN/nRnWBpG0Lh1SGQNcGVpRUs4cY6ZBCCKEcenISTCtxfsBdpIWxAbC9j6RJpPRHAPsBqwOPpFFUViENg36xeG3br0q6Mp/rp4QQQiilnmwAaS3O7zzg85I+UHgPOLZwzinAHravz8dtDFxDVQOY3zG+Bag5BFoUcYAhhNA9vdoATgFOhAGhEMdTiPPLKY9uAaYBB9g+Jwe43yPpFWBh/vpxbuw2BG7Ix15KGuqcK+ko4FBgDUnfBBYAL9LA8HKEQYRQW/xhGEZCTzaAVWmPKiEPCyrxfFXlGxfi/O6V9EIl/ZGktUk9w6m2189llVCIecB7bD8CnJq3zQCOtn1TB28vhBBCG/TyJJiWUx5V2H6SlPfviHwMNBYKEUIIoeR6ugGktZRHA9h+mPSc1s5Fg4VCDCnCIEIIoRx6vQEcVihEgaChUIghRRhECCGUQ0++A4RhpTyqPs8mpMkwT5KWRhsyFCKEEEL59WwDyDBSHhXK1wJOB06zbUkNhUI0KsIgQgihe0ZVA1gvu0MhpGFxdgfSsOYuknYCliWFJzxMVcqjHNKwJekd3znACnlJs9cBq5HW9FxH0nOkUIgTJB1t+ybbj0iaK2kn238Elidlgviy7UoAfV0RBhFCffHHYei00fYOsOHsDjkU4iXbE21vA7wLWBm40fbpMCCkwcDh+bilSb2+20hpj9YlrerydCUUosj2DrnxA3iMlFEihBBCyY2aBnCEQxq+ABxue24+9jnbZw9Rv71JPcxIhhtCCKPAqGkAGaGQhrwu6Mr5fA2RtCLwH8BXG9g3wiBCCKEERlMDOFIhDSINiTbjq6RsEPOG2jHCIEIIoRxGxSSYkQxpsP1FSS9I2iT3GBuxE/AhSSeRJs4skvSy7dOGc98hhBA6p6EGUNKmwGzb8yVNBrYjhRg828nKFYx0SMMJwPck7Wd7rqRK4zi9VuVs71K4zjRgXiONX4RBhBBC9zQ6BHohsFDSZsAZwMakRaJHyhTgYkn7SLKk1+c6vTmXb5/Ld8/lB+TjVpT0kqSXgb8ALwNfkzSBnN1B0qWSrs+LWs/NYRM/IPUU5+RjnyDNFgWYCFwlaXb++pmkHSXNzOEThwJNrQ4TQghh5DU6BLrI9oIch3eK7e9KurWTFSuqZHeQdAE5DML2NJZkYTgpl0+xPRWo5POrldnhy7a/AqxfzOwgaWPbO+R9DyXFDq6Ve4CrknqVkPL8Dcj4IGksMCk/o3WB2yQtY3vBYPcVcYAhDC5GSEInNdoDfDUPGX4M+GUuW7YzVaqtzGEQtl8sNHZjaH4STQghhBHWaAP4cWBn4Pi8+snGwEh3XUobBpGP20nSXcAdwKFD9f5CCCF0V0MNoO27SXFut+TvH7F9YicrVkOZwyCw/ce84sybgM/X64lGHGAIIZRDQw2gpPeT3n1dmb+fKOmywY9qn0IYxP9KmgV8FthP0tKkYcwv5/LvAnvmXlyt8xTDIPZjSRjELGAC6d3iXOCFvG/TbN8DvECdiTARBxhCCOXQ6CSYacCOpCFGbM/Mw6AjpdRhEPnYv+ZJMBuRFteeNdRNRRhECCF0T6MN4ALbzy2ZOwKM7ESPKUD1kOuFwJ4UMjsUyg9jYGaHSjaIc4D/KoZBVA4qZnYghUGsBPxJ0qvAq8C3C9e4PJcDXE+adXpMLltEmkDz1LDuOIQQQkc12gDeKekAYGlJmwNHAdd1rloDFcIgiumQTmVJGEQxHdJlLAmDqLYon28WsH4+9lJS1oedK2EQuXwRSxr5pRg4XPzeqjCId5FiDJcjDbE29HIvwiBCaI8YSQmtaHQW6JHANsB8Uizdc8C/dapSg2g4HVJWnQ7pPcBXKhsLcYCrFYd0cxzgu4AdbW9LSoc0oPtb5Sng/bbfQAoVOWeQfUMIIZTAkD3APNHkMtvvBI7tfJXq1qMSB7grqYc3LZdX4gDfBfxO0hjbL1cfb/tJSQeThjWn2TZL4gCfIDWqJ+TdvwDsWowDBAaLAywuCnAXMEbS8rbnD+OWQwghdNCQPUDbC4EX82oo3VTqOMCCDwK31mv8IgwihBDKodF3gC8Dd0i6hjTFHwDbR3WkVrVNAU7Jnyvxfrfw2jjAg0jvCeupFQdoSZU4wL/Q4gQfSdsA3wTeXW+fPJN0OsDEjTaJFWNCCKFLGm0AL89fXaHyp0NC0njSjNSPDqP3GEIIYYQ01AAOtg7mCCl7HOBqpD8QPm/7D43eVMQBhhBC9zSaD/ARagwL2m5ptZQWDCcO0MCDwIssiQN8DymV0uKlWHIc4KaS7ifF/a0OPCjpSQbGAW4LXJfLIcUB3ga8HjhXUiWu4Wjb3xvspiIMIoT2iT8mQ7MaHQKdVPg8BtgXWKP91amtEgdYVbY4DrCqvBgHeCGwLnBtTp8EgFJS30rYxFWFw1+wvUXep5I+6Q85fVKlp/cSMAd4T84hWAmbmAHsWyN9UgghhBJqdDHspwtfj9o+hfROrrTKnD4phBBC9zW6GPYOha9JucdTc8HpEill2ESEQYQQQjk0OgRaXAdzAfAI8OH2V6etShk2EWEQIYRQDo02gJ+oDgkY4WwQTRkNYRMhhBC6q9EG8OekNTOry97Y3uq0TanDJioiDCKEELpn0AZQ0utJi2CvKumfCptWIc0GLZVCtog/AV8ulG9PWtB6LHBGoXwL0v19SNIXgLGS7iS991uXtALOejlsYn3gBkkLgTtIz26cpLeT0id9nPSucQFwDwOHjWuKMIgQ2i/+qAyNGqoHuCXwPmA14P2F8ueBT3aqUsNQyQpxre0ra5Q/ZPt0gDz783LgM7Z/kct2JYU4vB9Yy/ZncvmWwKzcE3zJ9sRcfi4wyfZvJf07MI/U89x+JG42hBBC6wZtAG1fClwqaefK0F9ZtZAt4gDg+krjB2D71/mYg4E/F8rvq3PZ3wHb5X1+mxPthhBCGAUafQd4q6R/JQ0XLh76tP3PHalVaxaHPUh6RtIOtm+hEPYgaQYp7OEi0oouN9c515nA1ZI+BFwLnG37geIOkpYhrURzZY3j68qN68EA49cY18yhIYQQ2qjRhLjnAK8Ddgd+A4wnDYOWSXV4w5QhyuuyPRPYBPhP0oo3f5K0Vd68gqSZwE2kEIgzap+l7rmn255ke9K4lVYZ+oAQQggd0WgPcDPb+0ray/bZks5j4BJiXdVK2AMpce076p3T9jxST/EiSYtIPcd7yFnmO3pDIYQQOq7RBvDV/N9nc/D334AJHalRa1oJezgP+Lyk99q+PG/bA3iUNMv1btt/l7QcsDVp1Zi2ijCIEELonkaHQKdLWh34EmmCyd3ASR2rVfOmUDsrxJvrlB9g+yXSSjG/lDRL0t2k9UGfJPUmn8lZMG4lDXdemMMmxkh6UNI9ki6QtI6ksZJmAQ8A20iaL+nwztxqCCGEdpDdv6txSbqA2tkiTgJ2JoVNTM1lY0jxf82ETcwf7PoTN9rE1xzztXbfVgh9LUZVep+km21PGnrPwTW6GPY6ks6QdEX+fmtJnxjuxbuphWwRNcMmbN9JakQfLZTfN1TjF0IIobsaHQI9izTpZb38/f3Av3WiQiOo2WwRQ4VN/Iek6yV9XdLmHax3CCGENmi0AVzT9gXAIgDbC0iLRI9mIxU2MUCkQwohhHJodBboCznUwACS3gw817FaddgIh01U7xfpkEIIoQQabQA/Q5r9uamkPwBrkd6TjValCD2XjWEAABm/SURBVJuIMIgQQuieQYdAJW0IkJcUewfwFuAQYBvbt3e+eh3TatjE+4AjJT1QFTaxKfAbSXdQCJvoXPVDCCEM16BhEJJusb1D/nyh7Q+OWM1GQCF90la27y2Ub0/KHr+H7auqjlkLeAw4wvZ/F8r/Gfg0aZh4KeDYvJh4XREGEUJnxQhLbxqpMAgVPm8y3IuVUCVN0v51ymtNgNkXuKG4TdJ44Fjgbba3I/UkR3MPOYQQet5QDaDrfB71WogDrJgC/DswXtL6uWxt0uLg8yBNiLH9SEdvIIQQwrAM1QD+g6S5kp4Htsuf50p6XtJon8PfbBwgkjYAXmf7RuACYL+86TbgCeARST+UVEwePECEQYQQQjkM2gDaXtr2KrZXtr1M/lz5frTn8mklDnB/UsM3YJvthcAepJ7j/cDJkqbVumikQwohhHJoNAyip7QSB2j7eVKDt46kA/Op1pO0ue0HnGYT3QjcKOka4IfkrPQhhBDKpy8bQFqIA5R0I7Ci7fUL274K7C/pDNLQ6C1500Tgz0NVIuIAQwihe/q1AZwC/Db3/iohEBcCewI35fJKCMSFwGGkWL8rJb3KkhCIC0lDoWsDn8xB8LeTMkQcOlQlFsx5hjk/+HH77y6EUFf80RkqGl0LtKfYngy8nkIIhO1Tbe8JLEshBML2Zbb3zOmS7qMQAmH7dttbA+cCGwMv2p5o+115Ek0IIYSS6ssGsM0hENi+wfbjna53CCGE9unLBpD2hkA0JcIgQgihHPq1AWxbCESzIgwihBDKoe8mwXQiBKILtxFCCGGY+q4BpM0hEMBxrVYkwiBCCKF7+rEBnAKcCAOyQRxPIRVSIRvENOAAUgjExTn/33dIvcZfAu8GjpN0EvBJYMUcJnGj7bcOVZEIgwihnOIP0/7Qdw1gDoGoqGR9WJBDIKrLN66US1qatMzZu4DZwJ9YkhT4CmAH4L2250tau6M3EUIIYdj6dRJMK6EQOwIP2n7Y9iukiTB75W2HASfang9g+8kRuYkQQggt69sGkOZDIdYH/lo4fnYuA9gC2EXSHyX9RtKb6l00wiBCCKEc+rkBbDYUopgcuKKSI3EZYHXSe8TPAhfknuRrD4gwiBBCKIW+ewcIrYVCkHp8GxROMx54LH+eDVxUyQghaRGwJmlN0BBCCCXUlw0gLYRCAD8BNpe0MfAo6b3hAXm3S0gN6gxJWwDLAU8NVYkIgwghhO7p6SFQSftIsqTXV236CPBBSacVyi4GPgpsL2lm/jqWlPHhAOBVYCZwFXAPqfe3b97nMOCI3JucCcwHjuzkvYUQQhieXu8BVsIZ9mdgctpbGDihBeB1wO+AQ22/nIc9/932ZcBlkl4GtgfeZPspSUcDK9k+nhRHiKR5tldqtHIRBxhCb4qRndGhZ3uAg4Q5vBFYB7i6UDaWFMh+pO2XAWw/n1MgVSwApgOf7njlQwghdFzPNoDUCHOQtBTwbdJMzaLNgL/kNT8H8z3gQEmrtr+6IYQQRlIvN4C1whkOB/6f7erhzwEkfTy/A/xrToMEgO25wI+Ao1qtVMQBhhBCOfTkO8B6YQ6kd3y7SDocWAlYTtI84GvAhpXMD7Z/CPxQ0p352KJTSO8Qf9hK3WxPJw2lMnGjTTzE7iGEEDqkJxtA6oc5nG77wPz9VGCS7WPy92cAp0k6JE+CWZoUzjCA7WckXUB6t3jmcCoZYRAhhNA9vToEOoWc2aGgEs5Qz7HA48Cdkm4l9RbPZkmwe9G3SYHuIYQQRqme6wHmFEfvAA6t2nQWcI+khbaPsH2WpB9L+gawL/BC3u+HObRh8fkkXQRsVTjXCsDDtqfl4dZrc/k8SY+yZAWYHfPC2TVFGEQI/SdGfcqjF3uAxdi/ouOA31SVfR1YD3iD7YnALsCyDZ4PANtP256Yjz8dOLny/WCNXwghhO7qqQaw3bF/9c4XQghh9OupBpD2x/7VS5nUsgiDCCGEcui1BrDdsX/1UiO1LNIhhRBCOfTMJJh2x/4NkjLpcyN/dyGEENqtZxpA2h/7V+98b+O1C2m3JOIAQwihe3qpAZwCnFhVdiFwrKTdWRLGsKak64FVgIXAw6TYv+eBTUi9xscq55N0CfDOnOXh1/nrfmCr3JN8Np93Tr7Gw8C3GqlwhEGE0L/ij9/u65kG0PbkGmWnSnobOYwhx+1dlzb5AUnrATcDW9l+VtIMYA1S/N5kSasBXy2c8izgQ7a3lbQM8CvgFNsXSdoNGAscQgghhNLrtUkwA9QKY7B9v+0H8ufHgCeBtQqH/ZQlIQ//BFxU69y2FwDXkWaTYvtaYKhsEiGEEEqipxtAhghjkLQj6Z3fQ4Xia4G35/eB+wPn1zpxjiPcDbijmQpFGEQIIZRDrzeAdcMYJK0LnAN83PaiwjELSUOm+wEr2J5Vdc5NJc0E/gBcbvuKZioUYRAhhFAOPfMOsNoQYQwrA5cDX7R9Q43Df0paTHtajW0P5WXPQgghjGI92wBSP4zh7cCX87af1Tn2d8AJwE86WcEIgwghhO6R3Zs5WfOMzhNJmRsq2RzeDfwHsC4wP+/6BLC37ZmSngVeBtZ1fjA5DGIv28ozSv/P9piqa+1LmiE6Np/3KeATtq8arI4TN9rE1xzztTbcbQhhtIk/flsn6Wbbk4Z7np59B2h7su0rKWRzsH0qsCuwpe0VgE2B5YFZ+bCZpFmhbwXIYRDrsiRV0mzgwRqXuxPYgZRt4m22xw/V+IUQQuiunm0AobNhEEW277F9XxurHkIIocN6ugGkg2EQrYowiBBCKIdebwA7EQYxLBEGEUII5dCzs0A7GAYRQgihB/RsA0iEQYQQQhhELw+BTiH14oouJIUrvB2YmhPgzpRUDGyfDCwCfmn7qbxtBUl3AVcAr5c0O3/NlzRH0j6SZgM7A7+XtKDD9xZCCGGYejYOsFWSLiCFPlybs0dswdDZIw63/fscNnEVsE1OnzSoiAMMIbRDv40kRRxgB4xU2EQIIYTuiwZwoNKFTYQQQuiMaAAH6njYRMQBhhBCOfTyLNCmjFTYhO3pwHRI7wDbU/sQQgjNigZwiREPm4gwiBBC6J5oAJeYQsoeUVQJmxgPjJM0NZdPtT2zslPOHPGtEahjCCGENumLBlDSPuSUSLbvzbF9PwBWIb3DO9725LzvDOBo2zfl7BGnSpoMXAo8TEp5dKqkkwrHvB04BdiOlHViyBAIgAVznmHOD37ctvsMIfS3GFFqTr9MglmcEil//yLwUdvbAHsAp+QYvsH8zvb2trcEjgJOk7Rb3vYXYCpwXttrHkIIoSN6vgFsMbZvUHn482vAEfn7WbZvJ60gE0IIYRTo+QaQ1mL7GnEL8PpmKxNhECGEUA790AC2EtvXCLVSmUiHFEII5dDTk2CGGds3lO2Be9pW2RBCCCOqpxtAhhfbV5ek7YAvAf8ynMpFHGAIIXRPrzeAjcT2fQqYAOxj+5K8zx8kLQvMB34JfA/YRdKtwIakUIgngRMkfQV4CrieNKT8EUmvArvZvm6wykUYRAihTPrtD/KebgArcXpVZacCp1a+z+mPZgMTgUtsT87hDWOBQ2zvm3ddVdKJpFRJB9ueL2kd4B22r8j5ACfZfqqzdxVCCKEd+mESTF21QiQAbF8LPF+171jgk8CRtufn/Z6wfcHI1TiEEEK79HUDyBAhElU2A/5ie7DYhV/nDPN/rLdDhEGEEEI59HsDWDdEokW72p5oe6d6O0QYRAghlENPvwMczGAhEnlx62oPAhtKWtn28zW2hxBCGEX6tgGkfojE20jpjQaw/aKkM0gLYR9i+5UcSL+b7ZamckYYRAghdE+pG8DqLA657ErgzcDvbb+vsO8M0gzNl4DlgZNz8lnybM2T83F/B14hBcIfXXXJC4EDJH2HFOj+Sp7d+QnbVwHzgH/O+ywiLZ/275KOANYH5khaq9GZoBEGEUIoq37447zUDSADszhMy2X/SQ5RqLH/gbZvkrQG8JCks4BXgUuAs20fACBpI+ADtq8sHpxDJJB0EvAC8JDtqblsZ2BPYP0cArEmsJztxyQ9CWwOzGjTfYcQQuiw0k6CaSZEoYaVSA3YQtJ7vldsn144x59tf7fOdUUaHp0KvFvSmLxpXeCpQgjEUzmTBLZvtT2r2XsMIYTQPaVtAGkuRKHiXEm3A/cBx9leCGxDytzQqLcCj9h+iNSje08uvxrYQNL9kr4v6R1NnHOxCIMIIYRyKHMD2EqIwoG2tyMtV3Z0HuocQNL3JN0m6U/NXNf2POCNwMHAHOB8SVMbvJfFIgwihBDKoZTvAFsIURjA9hxJtwA7AXcBHyxs+9f8/u6mfK0fkia8PAa8P+/7AUnHklIejauEPuQe5QxghqQ7gI+R1hUNIYQwypSyAaTJEIVqedmy7YGTSMOf35B0mO0f5F3GVva1/fHCcbsDt9nevVB2NrC3pBuBRZVM8qS1Q//c4v0BEQYRQgjdVNYh0CnAxVVllRCF3wE/A3aTNDs3WhXnSpoJ3AycZfvm3GPcG3iHpEdyQ3Y28HNJllTM6n4mKevDL6uuewpwFTBT0vwcGrE1ME3SOvmd3gJgI+AxSde060GEEELoDDUwotiTchaIdYFrbU/LZcUsENUxhkcXQyyAdUghFteRQixOz/tWQixqzjItmrjRJr7mmK+19b5CCGGkdGsES9LNticN9zxl7QF2VLdCLEIIIZRHXzaAdC/EIoQQQkn0awPYrRCLiAMMIYSSKOss0I4ZyRCLOsdPB6ZDegc4rJsJIYTQsr5rABnBEIuhRBhECCF0Tz8OgY5EiMV/dP42QgghDEfHeoCSzgTeBzxpe9tC+c6klEIXACcCy5HSE33W9q9qnGcs8D/AdqSVWZ4F9rA9T9JC4I5cvhA4wvZ1gx0D7AKsJulbwCPAQbZPlTQhf39UZRanpNMkrWt7cv5+GeBvpFRKANh+nDyTVNJ3gY/b3rWRZxTpkEII/agsI1+d7AGeBexRo3wP4ErgKeD9tt9AWlLsnDrn+RTwhO035Ib0E6T4O4CXbE+0/Q/A54ETmjhmW+AZ4F8L13oS+JSk5erU5d2kWaAfzlkjFpM0CVitznEhhBBKpmMNoO3fkhqYarsB/5dTCD2Wy+4Cxkhavsb+6wKPFs57XyUlUZVVSMlumznmelIi24o5wLWkBrmWKcB3gL+QkusCIGlpUp7Cz9U5LoQQQsmM6CSYPEPyVdvPVW36IHBrnUbqTOBqSR8iNU5nF9bjXCG/lxtDavT+sYFjKnVZmtQYn1F1vROBK/IQbnH/FfL+h5B6elNIDSjAEcBlth+v6hjWegYHkzJKAMxf+/CD7hz0gO5bk9RbL7OoY3tEHdtnNNSze3U8/KBG96xXx9eEobXEdse+gAnAnYXvDwA+X7XPNqSlxTYd5DwrAf8EfJ/0Pm+rXD6vsM/OpJ6khjhmITAzl10LLF1dV+BHwEHAacDUXLYvcG7+PA74KymEYj1S1vplquvUwPO5qZPPv03/hlHHqGNpvkZDHUdLPaOOHvFZoHuS3v8BIGk8aUbmR50S0CJpH0kz89ckSLn4bF9k+3DgxyxJUruY7etJfy2sNcQxL9meSPoLYjkGvgOs+AZpJmfx+UwB3ilpFmkm6DhgV1JIxGbAg3nbWEkPNv1kQgghjKgRawDzpJHtSL0vJK0GXE7qEf6hsp/ti50mqUx0Wnz6rZJWz8csR8rC8Jo0RDmrw9LA040c4zQMexRpVZdlq7bdC9xNmsWKpFVIcYIb2p5gewKp4Zxi+3LbryuUv2h7s2E9rBBCCB3XsQZQ0k9I78i2zOmDPkd6z1dZ/eQIUs/pS4Ue39o1TrUp8JucgPZW0iorF+ZtK1SOBc4HPua0Rudgxyxm+1bgNgoLYhccD4zPn/8J+JUHvqO8lJQ4t9bEnUZNH8axIyXq2B5Rx/YYDXWE0VHPvq/jiKVDkvRF4EHbPx1y5xBCCKHD+jYfYAghhP7Wj0uhhRBCCNEADoekPSTdJ+lBScfU2C5Jp+bttxfzDtY7VtIakq6R9ED+7+qFbZ/P+99XtU5pKeooaYKklwrvdE+vvt4I1nFfSXdJWlSZTVzYVpbnWLOOJXuO/ynp3rz/xUqT1yrbyvIca9ax1efYwXoel/edKelqSesVtpXlWdasY5l+Jgvbj5ZkpfjySllzz7HbcR6j9Ys04/QhYBNSOMVtwNZV+7wHuIK0HumbgT8OdSwpy8Qx+fMxwDfz563zfssDG+fjly5ZHSdQiPvs8nPcCtgSmAFMKpyrTM+xXh3L9BzfzZIY12+W9OexXh2bfo4drucqheOPAk4v4bOsV8fS/Ezm7RsAV5Fm96/Z6nOMHmDrdiRN6nnY9iukxLp7Ve2zFyn1km3fQFqEe90hjt2LlFGC/N+9C+U/tT3f9iPAg/k8ZapjKzpSR9v32L6vxvVK8xwHqWMrOlXHq20vyMffwJKZ0WV6jvXq2KpO1bOYAXtFwIVzleVZ1qtjKzr1+wfgZFJkgavO1dRzjAawdeuTVoOpmM3AdUUH22ewY9dxyjBB/m8lNKSR63W7jgAbS7pV0m8k7TJE/TpZx+Fcr9t1hHI+x38m/bXe6PW6XUdo/jl2tJ6Sjpf0V+BA4MtNXK/bdYSS/ExK+gDwqO3bWrjeANEAtq7Wop/Vfy3V26eRY1u5XivHtLOOj5MWC9ge+AxwntIiAmWqYzzHFuoo6VhgAXBuE9frdh1beY4draftY21vkOt4RBPX63YdS/EzqZTq7lgGNszNXG+AaABbN5s0Dl0xHniswX0GO/aJPARA/u+TTVyvq3XMQw9P5883k8bgt+hSHYdzva7WsWzPUdLHSKsiHWi78gulVM+xVh1bfI4drWfBeaQkAI1er6t1LNHP5Kak93u3KS09OR64RdLrmrivJdzkC+L4WvwSdhng4fyPUXlJu03VPu9l4AveG4c6lpRWqTjB5KT8eRsGvuB9mKFflI90HddiyeLim5BSUq3RjToWjp3BwAkmpXmOg9SxNM+RlL/zbmCtqnOV5jkOUsemn2OH67l54fgjgZ+X8FnWq2Npfiarjp/FkkkwzT/HoX4Y4mvQf+D3APeT/ho6NpcdChyaPwv4Xt5+BwN/yb3m2Fw+jpSl4oH83zUK247N+98H7Fm2OpL+Wrwr/xDeQkp43K067kP6i3A+8ARwVQmfY806luw5Pkh6rzIzf51ewudYs46tPscO1vNC4E7gduAXwPolfJY161imn8mq888iN4CtPMdYCSaEEEJfineAIYQQ+lI0gCGEEPpSNIAhhBD6UjSAIYQQ+lI0gCGEEPpSNIAh1CFpYWH1+5mSJrRwjr0lbd3+2i1eof/OTpx7kGtOlPSekbxm4dpL5cwBd0q6Q9KfJG3cjbqE3rBMtysQQom9ZHviMM+xN/BLUqB2QyQt4yWLO5eGpGWAicAk4P91oQr7AesB29leJGk88MJwTljWZx1GRvQAQ2iCpDfmxYBvlnRVYUm4T+YeyW2SLpQ0VtJbgA8A/5l7kJtKmqGc+0/Smnk5JyRNlfQzSb8Arpa0oqQz8zlvlVS9in51vaZKukTSLyQ9IukISZ/Jx94gaY283wxJp0i6Lvekdszla+Tjb8/7b5fLp0maLulq4EfA14D98v3sJ2nHfK5b83+3LNTnIklXKuWNPKlQ1z0k3ZKf1bW5rJH7XRd43PYiANuzbf99kHM2dE+S1sr/Zn/KX29t9ucijFKNrowQX/HVb1/AQpasLnIxsCxwHXnJLVKP5Mz8eVzhuK8DR+bPZwEfKmybQV7tAlgTmJU/TyWtClNZVecbwEfy59VIK2KsWFW/CeQcbfn4B4GVSctWPceS1TZOBv6tcP3/yZ/fXjj+u8BX8ud/BGbmz9OAm4EVCtc5rVCHVViSi++dwIWF/R4GVgXGkPK2bZDr9ldg47xfM/c7nrTyx0zg28D2ubzeORu9p/OAt+XPGwL3dPtnL75G5iuGQEOob8AQqKRtgW2BayRBStr5eN68raSvk355r0RK1tmsa2w/kz+/G/iApKPz92PIv5wHOf7Xtp8Hnpf0HGkpK0hLTG1X2O8nALZ/K2kVpQzqb2PJwse/kjRO0qp5/8tsv1TnmqsCZ0vanLTy/rKFbdfafg5A0t3ARsDqwG+d8rXRzP3anp17mP+Yv66VtC8wts45G72ndwJb539TgFUkrZyfZehh0QCG0DgBd9neuca2s4C9bd8maSowuc45FrDk1cOYqm3F91kCPujmEubOL3xeVPh+EQP/X69e/9AMnkpmsPdsx5Ea3n3yJKEZdeqzMNdBNa4PDd6v7fmkxZOvkPQE6R3rNYOc8zWnyP8t3tNSwM6DNPKhR8U7wBAadx+wlqSdASQtK2mbvG1l4HFJy5ISiVY8n7dVzALemD9/aJBrXQUcqdwtkbT98Ku/2H75nG8Dnsu9tN+S6y1pMvCUB2YHr6i+n1VJmQEgDXsO5XrgHZXZm5V3kzRwv5J2kLRe/rwUqVf750HO2eg9Xc2SvHdIGu7EpzBKRAMYQoNsv0JqtL4p6TbSu6i35M1fAv5I6o3cWzjsp8Bn88SOTYFvAYdJuo70DrCe40jDibcrhToc18Zb+Xu+/unAJ3LZNGCSpNuBE4GP1Tn216ThwpmS9gNOAk6Q9AfSkPCgbM8BDgYuys/w/LypkftdG/hF3n47qTd92iDnbPSejqrsl4dqDx3qPkJviGwQIfQRSTOAo23f1O26hNBt0QMMIYTQl6IHGEIIoS9FDzCEEEJfigYwhBBCX4oGMIQQQl+KBjCEEEJfigYwhBBCX/r/5Q7G9UNr5VIAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "# Creating a bar plot\n",
    "sns.barplot(x=feature_imp, y=lists)\n",
    "# Add labels to your graph\n",
    "plt.xlabel('Feature Importance Score')\n",
    "plt.ylabel('Features')\n",
    "plt.title(\"Visualizing Important Features\")\n",
    "plt.legend()\n",
    "plt.ylim(0, 30)\n",
    "plt.show()\n",
    "#figure out why it is alphabetical#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n"
     ]
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "independentVariables = clean_patient_train\n",
    "rfModel_death = RandomForestClassifier()\n",
    "rfModel_death.fit(independentVariables, df_patient_info_train['FactorValue..death.from.disease.'])\n",
    "\n",
    "generatedDeathValues = rfModel_death.predict(clean_patient_test)\n",
    "print(generatedDeathValues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0. 0. 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0.\n",
      " 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 0. 1. 1. 0. 0. 0. 1. 0. 0.\n",
      " 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 1. 1. 1. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 1.]\n"
     ]
    }
   ],
   "source": [
    "independentVariables = clean_patient_train\n",
    "\n",
    "rfModel_risk = RandomForestClassifier()\n",
    "\n",
    "rfModel_risk.fit(independentVariables, df_patient_info_train['FactorValue..high.risk.'])\n",
    "generatedRiskValues = rfModel_risk.predict(clean_patient_test)\n",
    "print(generatedRiskValues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0. 0. 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0.\n",
      " 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 0. 1. 1. 0. 0. 0. 1. 0. 0.\n",
      " 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 1. 1. 1. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1.\n",
      " 0. 0. 0. 0. 0. 0. 0. 0. 1.]\n"
     ]
    }
   ],
   "source": [
    "independentVariables = clean_patient_train\n",
    "\n",
    "rfModel_progression = RandomForestClassifier()\n",
    "\n",
    "rfModel_progression.fit(independentVariables, df_patient_info_train['FactorValue..progression.'])\n",
    "generatedProgressionValues = rfModel_progression.predict(clean_patient_test)\n",
    "print(generatedRiskValues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1' '4' '4' '1' '1' '4' '2' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4'\n",
      " '4' '1' '1' '4' '4' '1' '4' '4' '4' '4' '1' '4' '4' '4' '4' '1' '4' '4'\n",
      " '4' '4' '4' '1' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4' '4'\n",
      " '4' '4' '4' '4' '4' '4' '1' '4' '4' '4' '4' '4' '4' '1' '4' '4' '4' '4'\n",
      " '4' '4' '1' '4' '4' '4' '4' '4' '1' '4' '4' '4' '4' '4' '4' '4' '4' '1'\n",
      " '4' '4' '4' '1' '4' '4' '4' '1' '1' '4' '4' '4' '4S' '4' '4' '4' '4' '4'\n",
      " '1' '4' '4' '2' '4' '4' '4' '1' '4' '1' '4' '4' '4' '1' '4' '4' '4' '1'\n",
      " '4' '1' '1' '1' '4' '4' '1' '1' '1' '4' '4' '4' '3' '4' '4' '1' '4' '4'\n",
      " '4' '1' '4' '1' '1' '1' '4' '4' '4' '3' '4' '1' '1' '4' '4' '1' '1' '4'\n",
      " '1' '1' '1' '1' '1' '1' '2' '1' '1' '1' '4' '1' '4' '2' '1' '4' '4' '1'\n",
      " '4' '1' '4' '2' '1' '1' '2' '2' '1' '2' '1' '4' '1' '4' '2' '1' '4' '4'\n",
      " '4' '4' '1' '1' '1' '1' '4' '1' '1' '4' '4' '1' '1' '4' '4' '1' '1' '2'\n",
      " '2' '1' '1' '2' '2' '2' '1' '2' '2' '2' '1' '1' '1' '4' '4' '4' '4' '4'\n",
      " '3' '4' '1' '4' '1' '4' '4' '3' '1' '1' '1' '1' '4' '4' '4']\n"
     ]
    }
   ],
   "source": [
    "independentVariables = clean_patient_train\n",
    "\n",
    "rfModel_INSS = RandomForestClassifier()\n",
    "\n",
    "rfModel_INSS.fit(independentVariables, df_patient_info_train['FactorValue..inss.stage.'])\n",
    "generatedStageValues = rfModel_INSS.predict(clean_patient_test)\n",
    "print(generatedStageValues)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.34 0.18 0.08 0.33 0.07]\n",
      " [0.25 0.2  0.09 0.38 0.08]\n",
      " [0.19 0.06 0.06 0.6  0.09]\n",
      " ...\n",
      " [0.21 0.11 0.11 0.44 0.13]\n",
      " [0.14 0.06 0.1  0.55 0.15]\n",
      " [0.26 0.04 0.02 0.61 0.07]]\n"
     ]
    }
   ],
   "source": [
    "probs_INSS = rfModel_INSS.predict_proba(clean_patient_test)\n",
    "print(probs_INSS)\n",
    "# prints matrix of probabilities for each class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.73 0.27]\n",
      " [0.83 0.17]\n",
      " [0.58 0.42]\n",
      " [0.83 0.17]\n",
      " [0.92 0.08]\n",
      " [0.67 0.33]\n",
      " [0.74 0.26]\n",
      " [0.76 0.24]\n",
      " [0.74 0.26]\n",
      " [0.8  0.2 ]\n",
      " [0.79 0.21]\n",
      " [0.82 0.18]\n",
      " [0.8  0.2 ]\n",
      " [0.68 0.32]\n",
      " [0.72 0.28]\n",
      " [0.76 0.24]\n",
      " [0.93 0.07]\n",
      " [0.67 0.33]\n",
      " [0.74 0.26]\n",
      " [0.77 0.23]\n",
      " [0.73 0.27]\n",
      " [0.69 0.31]\n",
      " [0.7  0.3 ]\n",
      " [0.76 0.24]\n",
      " [0.77 0.23]\n",
      " [0.69 0.31]\n",
      " [0.72 0.28]\n",
      " [0.63 0.37]\n",
      " [0.82 0.18]\n",
      " [0.66 0.34]\n",
      " [0.71 0.29]\n",
      " [0.68 0.32]\n",
      " [0.7  0.3 ]\n",
      " [0.86 0.14]\n",
      " [0.89 0.11]\n",
      " [0.69 0.31]\n",
      " [0.69 0.31]\n",
      " [0.66 0.34]\n",
      " [0.63 0.37]\n",
      " [0.87 0.13]\n",
      " [0.76 0.24]\n",
      " [0.68 0.32]\n",
      " [0.84 0.16]\n",
      " [0.76 0.24]\n",
      " [0.72 0.28]\n",
      " [0.78 0.22]\n",
      " [0.82 0.18]\n",
      " [0.78 0.22]\n",
      " [0.73 0.27]\n",
      " [0.71 0.29]\n",
      " [0.71 0.29]\n",
      " [0.6  0.4 ]\n",
      " [0.79 0.21]\n",
      " [0.71 0.29]\n",
      " [0.61 0.39]\n",
      " [0.53 0.47]\n",
      " [0.55 0.45]\n",
      " [0.62 0.38]\n",
      " [0.7  0.3 ]\n",
      " [0.49 0.51]\n",
      " [0.76 0.24]\n",
      " [0.9  0.1 ]\n",
      " [0.7  0.3 ]\n",
      " [0.74 0.26]\n",
      " [0.66 0.34]\n",
      " [0.73 0.27]\n",
      " [0.79 0.21]\n",
      " [0.83 0.17]\n",
      " [0.84 0.16]\n",
      " [0.73 0.27]\n",
      " [0.6  0.4 ]\n",
      " [0.8  0.2 ]\n",
      " [0.69 0.31]\n",
      " [0.65 0.35]\n",
      " [0.81 0.19]\n",
      " [0.86 0.14]\n",
      " [0.79 0.21]\n",
      " [0.72 0.28]\n",
      " [0.7  0.3 ]\n",
      " [0.63 0.37]\n",
      " [0.85 0.15]\n",
      " [0.73 0.27]\n",
      " [0.79 0.21]\n",
      " [0.72 0.28]\n",
      " [0.84 0.16]\n",
      " [0.72 0.28]\n",
      " [0.7  0.3 ]\n",
      " [0.76 0.24]\n",
      " [0.54 0.46]\n",
      " [0.74 0.26]\n",
      " [0.81 0.19]\n",
      " [0.66 0.34]\n",
      " [0.8  0.2 ]\n",
      " [0.83 0.17]\n",
      " [0.65 0.35]\n",
      " [0.79 0.21]\n",
      " [0.91 0.09]\n",
      " [0.85 0.15]\n",
      " [0.88 0.12]\n",
      " [0.82 0.18]\n",
      " [0.72 0.28]\n",
      " [0.84 0.16]\n",
      " [0.81 0.19]\n",
      " [0.77 0.23]\n",
      " [0.76 0.24]\n",
      " [0.61 0.39]\n",
      " [0.63 0.37]\n",
      " [0.69 0.31]\n",
      " [0.76 0.24]\n",
      " [0.83 0.17]\n",
      " [0.78 0.22]\n",
      " [0.5  0.5 ]\n",
      " [0.79 0.21]\n",
      " [0.69 0.31]\n",
      " [0.73 0.27]\n",
      " [0.7  0.3 ]\n",
      " [0.76 0.24]\n",
      " [0.84 0.16]\n",
      " [0.84 0.16]\n",
      " [0.9  0.1 ]\n",
      " [0.78 0.22]\n",
      " [0.8  0.2 ]\n",
      " [0.89 0.11]\n",
      " [0.82 0.18]\n",
      " [0.82 0.18]\n",
      " [0.77 0.23]\n",
      " [0.86 0.14]\n",
      " [0.82 0.18]\n",
      " [0.88 0.12]\n",
      " [0.95 0.05]\n",
      " [0.89 0.11]\n",
      " [0.87 0.13]\n",
      " [0.88 0.12]\n",
      " [0.84 0.16]\n",
      " [0.92 0.08]\n",
      " [0.85 0.15]\n",
      " [0.78 0.22]\n",
      " [0.81 0.19]\n",
      " [0.89 0.11]\n",
      " [0.74 0.26]\n",
      " [0.73 0.27]\n",
      " [0.89 0.11]\n",
      " [0.77 0.23]\n",
      " [0.78 0.22]\n",
      " [0.76 0.24]\n",
      " [0.9  0.1 ]\n",
      " [0.72 0.28]\n",
      " [0.84 0.16]\n",
      " [0.83 0.17]\n",
      " [0.89 0.11]\n",
      " [0.79 0.21]\n",
      " [0.78 0.22]\n",
      " [0.83 0.17]\n",
      " [0.83 0.17]\n",
      " [0.82 0.18]\n",
      " [0.71 0.29]\n",
      " [0.93 0.07]\n",
      " [0.91 0.09]\n",
      " [0.8  0.2 ]\n",
      " [0.85 0.15]\n",
      " [0.79 0.21]\n",
      " [0.66 0.34]\n",
      " [0.92 0.08]\n",
      " [0.82 0.18]\n",
      " [0.81 0.19]\n",
      " [0.77 0.23]\n",
      " [0.85 0.15]\n",
      " [0.88 0.12]\n",
      " [0.85 0.15]\n",
      " [0.68 0.32]\n",
      " [0.7  0.3 ]\n",
      " [0.88 0.12]\n",
      " [0.88 0.12]\n",
      " [0.82 0.18]\n",
      " [0.44 0.56]\n",
      " [0.81 0.19]\n",
      " [0.87 0.13]\n",
      " [0.72 0.28]\n",
      " [0.9  0.1 ]\n",
      " [0.92 0.08]\n",
      " [0.67 0.33]\n",
      " [0.87 0.13]\n",
      " [0.78 0.22]\n",
      " [0.8  0.2 ]\n",
      " [0.81 0.19]\n",
      " [0.66 0.34]\n",
      " [0.7  0.3 ]\n",
      " [0.83 0.17]\n",
      " [0.86 0.14]\n",
      " [0.67 0.33]\n",
      " [0.8  0.2 ]\n",
      " [0.69 0.31]\n",
      " [0.82 0.18]\n",
      " [0.9  0.1 ]\n",
      " [0.87 0.13]\n",
      " [0.8  0.2 ]\n",
      " [0.76 0.24]\n",
      " [0.89 0.11]\n",
      " [0.81 0.19]\n",
      " [0.85 0.15]\n",
      " [0.86 0.14]\n",
      " [0.76 0.24]\n",
      " [0.86 0.14]\n",
      " [0.87 0.13]\n",
      " [0.78 0.22]\n",
      " [0.83 0.17]\n",
      " [0.78 0.22]\n",
      " [0.68 0.32]\n",
      " [0.75 0.25]\n",
      " [0.73 0.27]\n",
      " [0.82 0.18]\n",
      " [0.93 0.07]\n",
      " [0.85 0.15]\n",
      " [0.94 0.06]\n",
      " [0.77 0.23]\n",
      " [0.84 0.16]\n",
      " [0.71 0.29]\n",
      " [0.73 0.27]\n",
      " [0.78 0.22]\n",
      " [0.75 0.25]\n",
      " [0.85 0.15]\n",
      " [0.82 0.18]\n",
      " [0.81 0.19]\n",
      " [0.88 0.12]\n",
      " [0.84 0.16]\n",
      " [0.68 0.32]\n",
      " [0.92 0.08]\n",
      " [0.86 0.14]\n",
      " [0.79 0.21]\n",
      " [0.8  0.2 ]\n",
      " [0.78 0.22]\n",
      " [0.83 0.17]\n",
      " [0.9  0.1 ]\n",
      " [0.7  0.3 ]\n",
      " [0.9  0.1 ]\n",
      " [0.65 0.35]\n",
      " [0.82 0.18]\n",
      " [0.64 0.36]\n",
      " [0.9  0.1 ]\n",
      " [0.77 0.23]\n",
      " [0.79 0.21]\n",
      " [0.84 0.16]\n",
      " [0.87 0.13]\n",
      " [0.78 0.22]\n",
      " [0.89 0.11]\n",
      " [0.84 0.16]\n",
      " [0.71 0.29]\n",
      " [0.79 0.21]\n",
      " [0.76 0.24]]\n"
     ]
    }
   ],
   "source": [
    "probs_death = rfModel_death.predict_proba(clean_patient_test)\n",
    "print(probs_death)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.52 0.48]\n",
      " [0.8  0.2 ]\n",
      " [0.49 0.51]\n",
      " [0.67 0.33]\n",
      " [0.79 0.21]\n",
      " [0.42 0.58]\n",
      " [0.65 0.35]\n",
      " [0.69 0.31]\n",
      " [0.63 0.37]\n",
      " [0.52 0.48]\n",
      " [0.6  0.4 ]\n",
      " [0.59 0.41]\n",
      " [0.64 0.36]\n",
      " [0.49 0.51]\n",
      " [0.5  0.5 ]\n",
      " [0.55 0.45]\n",
      " [0.67 0.33]\n",
      " [0.65 0.35]\n",
      " [0.43 0.57]\n",
      " [0.66 0.34]\n",
      " [0.67 0.33]\n",
      " [0.55 0.45]\n",
      " [0.43 0.57]\n",
      " [0.73 0.27]\n",
      " [0.47 0.53]\n",
      " [0.5  0.5 ]\n",
      " [0.45 0.55]\n",
      " [0.67 0.33]\n",
      " [0.65 0.35]\n",
      " [0.43 0.57]\n",
      " [0.47 0.53]\n",
      " [0.49 0.51]\n",
      " [0.6  0.4 ]\n",
      " [0.69 0.31]\n",
      " [0.64 0.36]\n",
      " [0.52 0.48]\n",
      " [0.59 0.41]\n",
      " [0.54 0.46]\n",
      " [0.5  0.5 ]\n",
      " [0.67 0.33]\n",
      " [0.4  0.6 ]\n",
      " [0.42 0.58]\n",
      " [0.63 0.37]\n",
      " [0.63 0.37]\n",
      " [0.6  0.4 ]\n",
      " [0.42 0.58]\n",
      " [0.52 0.48]\n",
      " [0.58 0.42]\n",
      " [0.52 0.48]\n",
      " [0.47 0.53]\n",
      " [0.63 0.37]\n",
      " [0.44 0.56]\n",
      " [0.76 0.24]\n",
      " [0.48 0.52]\n",
      " [0.41 0.59]\n",
      " [0.35 0.65]\n",
      " [0.5  0.5 ]\n",
      " [0.59 0.41]\n",
      " [0.58 0.42]\n",
      " [0.37 0.63]\n",
      " [0.58 0.42]\n",
      " [0.67 0.33]\n",
      " [0.61 0.39]\n",
      " [0.57 0.43]\n",
      " [0.5  0.5 ]\n",
      " [0.61 0.39]\n",
      " [0.64 0.36]\n",
      " [0.6  0.4 ]\n",
      " [0.65 0.35]\n",
      " [0.56 0.44]\n",
      " [0.62 0.38]\n",
      " [0.52 0.48]\n",
      " [0.58 0.42]\n",
      " [0.6  0.4 ]\n",
      " [0.71 0.29]\n",
      " [0.74 0.26]\n",
      " [0.65 0.35]\n",
      " [0.62 0.38]\n",
      " [0.62 0.38]\n",
      " [0.5  0.5 ]\n",
      " [0.67 0.33]\n",
      " [0.32 0.68]\n",
      " [0.59 0.41]\n",
      " [0.53 0.47]\n",
      " [0.77 0.23]\n",
      " [0.66 0.34]\n",
      " [0.54 0.46]\n",
      " [0.43 0.57]\n",
      " [0.52 0.48]\n",
      " [0.51 0.49]\n",
      " [0.7  0.3 ]\n",
      " [0.46 0.54]\n",
      " [0.64 0.36]\n",
      " [0.63 0.37]\n",
      " [0.52 0.48]\n",
      " [0.6  0.4 ]\n",
      " [0.79 0.21]\n",
      " [0.67 0.33]\n",
      " [0.75 0.25]\n",
      " [0.79 0.21]\n",
      " [0.71 0.29]\n",
      " [0.76 0.24]\n",
      " [0.74 0.26]\n",
      " [0.77 0.23]\n",
      " [0.61 0.39]\n",
      " [0.66 0.34]\n",
      " [0.63 0.37]\n",
      " [0.62 0.38]\n",
      " [0.83 0.17]\n",
      " [0.69 0.31]\n",
      " [0.6  0.4 ]\n",
      " [0.65 0.35]\n",
      " [0.72 0.28]\n",
      " [0.55 0.45]\n",
      " [0.83 0.17]\n",
      " [0.7  0.3 ]\n",
      " [0.53 0.47]\n",
      " [0.65 0.35]\n",
      " [0.55 0.45]\n",
      " [0.68 0.32]\n",
      " [0.61 0.39]\n",
      " [0.68 0.32]\n",
      " [0.47 0.53]\n",
      " [0.47 0.53]\n",
      " [0.65 0.35]\n",
      " [0.75 0.25]\n",
      " [0.62 0.38]\n",
      " [0.74 0.26]\n",
      " [0.71 0.29]\n",
      " [0.85 0.15]\n",
      " [0.82 0.18]\n",
      " [0.79 0.21]\n",
      " [0.71 0.29]\n",
      " [0.63 0.37]\n",
      " [0.76 0.24]\n",
      " [0.79 0.21]\n",
      " [0.73 0.27]\n",
      " [0.69 0.31]\n",
      " [0.85 0.15]\n",
      " [0.62 0.38]\n",
      " [0.59 0.41]\n",
      " [0.8  0.2 ]\n",
      " [0.64 0.36]\n",
      " [0.71 0.29]\n",
      " [0.67 0.33]\n",
      " [0.81 0.19]\n",
      " [0.41 0.59]\n",
      " [0.71 0.29]\n",
      " [0.7  0.3 ]\n",
      " [0.91 0.09]\n",
      " [0.64 0.36]\n",
      " [0.79 0.21]\n",
      " [0.68 0.32]\n",
      " [0.85 0.15]\n",
      " [0.74 0.26]\n",
      " [0.76 0.24]\n",
      " [0.85 0.15]\n",
      " [0.82 0.18]\n",
      " [0.64 0.36]\n",
      " [0.67 0.33]\n",
      " [0.67 0.33]\n",
      " [0.63 0.37]\n",
      " [0.79 0.21]\n",
      " [0.64 0.36]\n",
      " [0.66 0.34]\n",
      " [0.78 0.22]\n",
      " [0.72 0.28]\n",
      " [0.88 0.12]\n",
      " [0.71 0.29]\n",
      " [0.69 0.31]\n",
      " [0.62 0.38]\n",
      " [0.72 0.28]\n",
      " [0.69 0.31]\n",
      " [0.7  0.3 ]\n",
      " [0.53 0.47]\n",
      " [0.82 0.18]\n",
      " [0.89 0.11]\n",
      " [0.54 0.46]\n",
      " [0.82 0.18]\n",
      " [0.65 0.35]\n",
      " [0.61 0.39]\n",
      " [0.82 0.18]\n",
      " [0.4  0.6 ]\n",
      " [0.67 0.33]\n",
      " [0.6  0.4 ]\n",
      " [0.74 0.26]\n",
      " [0.74 0.26]\n",
      " [0.75 0.25]\n",
      " [0.83 0.17]\n",
      " [0.71 0.29]\n",
      " [0.75 0.25]\n",
      " [0.48 0.52]\n",
      " [0.69 0.31]\n",
      " [0.66 0.34]\n",
      " [0.69 0.31]\n",
      " [0.62 0.38]\n",
      " [0.55 0.45]\n",
      " [0.71 0.29]\n",
      " [0.58 0.42]\n",
      " [0.56 0.44]\n",
      " [0.73 0.27]\n",
      " [0.7  0.3 ]\n",
      " [0.63 0.37]\n",
      " [0.73 0.27]\n",
      " [0.44 0.56]\n",
      " [0.66 0.34]\n",
      " [0.64 0.36]\n",
      " [0.5  0.5 ]\n",
      " [0.5  0.5 ]\n",
      " [0.58 0.42]\n",
      " [0.64 0.36]\n",
      " [0.76 0.24]\n",
      " [0.62 0.38]\n",
      " [0.88 0.12]\n",
      " [0.63 0.37]\n",
      " [0.81 0.19]\n",
      " [0.71 0.29]\n",
      " [0.79 0.21]\n",
      " [0.73 0.27]\n",
      " [0.76 0.24]\n",
      " [0.9  0.1 ]\n",
      " [0.73 0.27]\n",
      " [0.72 0.28]\n",
      " [0.75 0.25]\n",
      " [0.84 0.16]\n",
      " [0.81 0.19]\n",
      " [0.8  0.2 ]\n",
      " [0.83 0.17]\n",
      " [0.7  0.3 ]\n",
      " [0.69 0.31]\n",
      " [0.67 0.33]\n",
      " [0.66 0.34]\n",
      " [0.69 0.31]\n",
      " [0.7  0.3 ]\n",
      " [0.66 0.34]\n",
      " [0.69 0.31]\n",
      " [0.78 0.22]\n",
      " [0.65 0.35]\n",
      " [0.78 0.22]\n",
      " [0.54 0.46]\n",
      " [0.56 0.44]\n",
      " [0.67 0.33]\n",
      " [0.67 0.33]\n",
      " [0.79 0.21]\n",
      " [0.85 0.15]\n",
      " [0.69 0.31]\n",
      " [0.55 0.45]\n",
      " [0.75 0.25]\n",
      " [0.42 0.58]]\n"
     ]
    }
   ],
   "source": [
    "probs_risk = rfModel_risk.predict_proba(clean_patient_test)\n",
    "print(probs_risk)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[0.59 0.41]\n",
      " [0.71 0.29]\n",
      " [0.49 0.51]\n",
      " [0.71 0.29]\n",
      " [0.73 0.27]\n",
      " [0.54 0.46]\n",
      " [0.64 0.36]\n",
      " [0.73 0.27]\n",
      " [0.59 0.41]\n",
      " [0.58 0.42]\n",
      " [0.62 0.38]\n",
      " [0.67 0.33]\n",
      " [0.59 0.41]\n",
      " [0.47 0.53]\n",
      " [0.56 0.44]\n",
      " [0.58 0.42]\n",
      " [0.76 0.24]\n",
      " [0.61 0.39]\n",
      " [0.61 0.39]\n",
      " [0.51 0.49]\n",
      " [0.66 0.34]\n",
      " [0.51 0.49]\n",
      " [0.44 0.56]\n",
      " [0.7  0.3 ]\n",
      " [0.53 0.47]\n",
      " [0.55 0.45]\n",
      " [0.62 0.38]\n",
      " [0.5  0.5 ]\n",
      " [0.59 0.41]\n",
      " [0.54 0.46]\n",
      " [0.47 0.53]\n",
      " [0.46 0.54]\n",
      " [0.56 0.44]\n",
      " [0.69 0.31]\n",
      " [0.54 0.46]\n",
      " [0.54 0.46]\n",
      " [0.62 0.38]\n",
      " [0.61 0.39]\n",
      " [0.57 0.43]\n",
      " [0.67 0.33]\n",
      " [0.62 0.38]\n",
      " [0.55 0.45]\n",
      " [0.64 0.36]\n",
      " [0.54 0.46]\n",
      " [0.58 0.42]\n",
      " [0.6  0.4 ]\n",
      " [0.68 0.32]\n",
      " [0.6  0.4 ]\n",
      " [0.62 0.38]\n",
      " [0.57 0.43]\n",
      " [0.58 0.42]\n",
      " [0.48 0.52]\n",
      " [0.72 0.28]\n",
      " [0.62 0.38]\n",
      " [0.5  0.5 ]\n",
      " [0.49 0.51]\n",
      " [0.54 0.46]\n",
      " [0.54 0.46]\n",
      " [0.63 0.37]\n",
      " [0.55 0.45]\n",
      " [0.57 0.43]\n",
      " [0.7  0.3 ]\n",
      " [0.58 0.42]\n",
      " [0.78 0.22]\n",
      " [0.64 0.36]\n",
      " [0.71 0.29]\n",
      " [0.79 0.21]\n",
      " [0.75 0.25]\n",
      " [0.6  0.4 ]\n",
      " [0.67 0.33]\n",
      " [0.58 0.42]\n",
      " [0.62 0.38]\n",
      " [0.46 0.54]\n",
      " [0.69 0.31]\n",
      " [0.69 0.31]\n",
      " [0.63 0.37]\n",
      " [0.72 0.28]\n",
      " [0.59 0.41]\n",
      " [0.65 0.35]\n",
      " [0.56 0.44]\n",
      " [0.67 0.33]\n",
      " [0.56 0.44]\n",
      " [0.66 0.34]\n",
      " [0.6  0.4 ]\n",
      " [0.62 0.38]\n",
      " [0.6  0.4 ]\n",
      " [0.43 0.57]\n",
      " [0.63 0.37]\n",
      " [0.54 0.46]\n",
      " [0.52 0.48]\n",
      " [0.68 0.32]\n",
      " [0.63 0.37]\n",
      " [0.68 0.32]\n",
      " [0.64 0.36]\n",
      " [0.53 0.47]\n",
      " [0.73 0.27]\n",
      " [0.73 0.27]\n",
      " [0.64 0.36]\n",
      " [0.7  0.3 ]\n",
      " [0.57 0.43]\n",
      " [0.61 0.39]\n",
      " [0.58 0.42]\n",
      " [0.6  0.4 ]\n",
      " [0.59 0.41]\n",
      " [0.52 0.48]\n",
      " [0.57 0.43]\n",
      " [0.62 0.38]\n",
      " [0.68 0.32]\n",
      " [0.7  0.3 ]\n",
      " [0.53 0.47]\n",
      " [0.51 0.49]\n",
      " [0.58 0.42]\n",
      " [0.77 0.23]\n",
      " [0.58 0.42]\n",
      " [0.65 0.35]\n",
      " [0.64 0.36]\n",
      " [0.72 0.28]\n",
      " [0.69 0.31]\n",
      " [0.64 0.36]\n",
      " [0.78 0.22]\n",
      " [0.61 0.39]\n",
      " [0.6  0.4 ]\n",
      " [0.61 0.39]\n",
      " [0.59 0.41]\n",
      " [0.71 0.29]\n",
      " [0.69 0.31]\n",
      " [0.67 0.33]\n",
      " [0.7  0.3 ]\n",
      " [0.7  0.3 ]\n",
      " [0.77 0.23]\n",
      " [0.59 0.41]\n",
      " [0.73 0.27]\n",
      " [0.73 0.27]\n",
      " [0.61 0.39]\n",
      " [0.72 0.28]\n",
      " [0.74 0.26]\n",
      " [0.66 0.34]\n",
      " [0.62 0.38]\n",
      " [0.71 0.29]\n",
      " [0.55 0.45]\n",
      " [0.6  0.4 ]\n",
      " [0.78 0.22]\n",
      " [0.62 0.38]\n",
      " [0.71 0.29]\n",
      " [0.57 0.43]\n",
      " [0.78 0.22]\n",
      " [0.55 0.45]\n",
      " [0.56 0.44]\n",
      " [0.74 0.26]\n",
      " [0.75 0.25]\n",
      " [0.56 0.44]\n",
      " [0.73 0.27]\n",
      " [0.67 0.33]\n",
      " [0.73 0.27]\n",
      " [0.79 0.21]\n",
      " [0.77 0.23]\n",
      " [0.7  0.3 ]\n",
      " [0.68 0.32]\n",
      " [0.63 0.37]\n",
      " [0.65 0.35]\n",
      " [0.68 0.32]\n",
      " [0.56 0.44]\n",
      " [0.78 0.22]\n",
      " [0.64 0.36]\n",
      " [0.68 0.32]\n",
      " [0.56 0.44]\n",
      " [0.82 0.18]\n",
      " [0.74 0.26]\n",
      " [0.72 0.28]\n",
      " [0.7  0.3 ]\n",
      " [0.6  0.4 ]\n",
      " [0.71 0.29]\n",
      " [0.66 0.34]\n",
      " [0.67 0.33]\n",
      " [0.51 0.49]\n",
      " [0.82 0.18]\n",
      " [0.87 0.13]\n",
      " [0.56 0.44]\n",
      " [0.74 0.26]\n",
      " [0.72 0.28]\n",
      " [0.58 0.42]\n",
      " [0.63 0.37]\n",
      " [0.62 0.38]\n",
      " [0.72 0.28]\n",
      " [0.72 0.28]\n",
      " [0.6  0.4 ]\n",
      " [0.61 0.39]\n",
      " [0.78 0.22]\n",
      " [0.65 0.35]\n",
      " [0.59 0.41]\n",
      " [0.67 0.33]\n",
      " [0.56 0.44]\n",
      " [0.57 0.43]\n",
      " [0.73 0.27]\n",
      " [0.64 0.36]\n",
      " [0.68 0.32]\n",
      " [0.56 0.44]\n",
      " [0.62 0.38]\n",
      " [0.65 0.35]\n",
      " [0.56 0.44]\n",
      " [0.71 0.29]\n",
      " [0.61 0.39]\n",
      " [0.74 0.26]\n",
      " [0.64 0.36]\n",
      " [0.6  0.4 ]\n",
      " [0.65 0.35]\n",
      " [0.59 0.41]\n",
      " [0.53 0.47]\n",
      " [0.58 0.42]\n",
      " [0.51 0.49]\n",
      " [0.68 0.32]\n",
      " [0.81 0.19]\n",
      " [0.71 0.29]\n",
      " [0.76 0.24]\n",
      " [0.58 0.42]\n",
      " [0.73 0.27]\n",
      " [0.7  0.3 ]\n",
      " [0.62 0.38]\n",
      " [0.64 0.36]\n",
      " [0.61 0.39]\n",
      " [0.86 0.14]\n",
      " [0.71 0.29]\n",
      " [0.8  0.2 ]\n",
      " [0.72 0.28]\n",
      " [0.76 0.24]\n",
      " [0.67 0.33]\n",
      " [0.79 0.21]\n",
      " [0.7  0.3 ]\n",
      " [0.66 0.34]\n",
      " [0.76 0.24]\n",
      " [0.65 0.35]\n",
      " [0.61 0.39]\n",
      " [0.78 0.22]\n",
      " [0.64 0.36]\n",
      " [0.66 0.34]\n",
      " [0.58 0.42]\n",
      " [0.59 0.41]\n",
      " [0.6  0.4 ]\n",
      " [0.78 0.22]\n",
      " [0.72 0.28]\n",
      " [0.66 0.34]\n",
      " [0.58 0.42]\n",
      " [0.67 0.33]\n",
      " [0.81 0.19]\n",
      " [0.75 0.25]\n",
      " [0.73 0.27]\n",
      " [0.51 0.49]\n",
      " [0.73 0.27]\n",
      " [0.63 0.37]]\n"
     ]
    }
   ],
   "source": [
    "probs_progression = rfModel_progression.predict_proba(clean_patient_test)\n",
    "print(probs_progression)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'bootstrap': True,\n",
       " 'ccp_alpha': 0.0,\n",
       " 'class_weight': None,\n",
       " 'criterion': 'entropy',\n",
       " 'max_depth': 1,\n",
       " 'max_features': 'auto',\n",
       " 'max_leaf_nodes': None,\n",
       " 'max_samples': None,\n",
       " 'min_impurity_decrease': 0.0,\n",
       " 'min_impurity_split': None,\n",
       " 'min_samples_leaf': 5,\n",
       " 'min_samples_split': 2,\n",
       " 'min_weight_fraction_leaf': 0.0,\n",
       " 'n_estimators': 50,\n",
       " 'n_jobs': None,\n",
       " 'oob_score': False,\n",
       " 'random_state': None,\n",
       " 'verbose': 0,\n",
       " 'warm_start': False}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.model_selection import LeaveOneOut, GridSearchCV, KFold, StratifiedKFold\n",
    "\n",
    "parameters = {\n",
    "'criterion': ['gini','entropy'],\n",
    "'n_estimators': [20,30,50,100],\n",
    "'max_depth':[1,2,3,4,5],\n",
    "'min_samples_leaf':[2,5,7,10],\n",
    "}\n",
    "\n",
    "\n",
    "rfModel_death = RandomForestClassifier()\n",
    "\n",
    "rf_grid_search = GridSearchCV(rfModel_death, parameters, cv=5,scoring='roc_auc_ovo')\n",
    "grid_search = rf_grid_search.fit(independentVariables, df_patient_info_train['FactorValue..death.from.disease.'])\n",
    "\n",
    "best_random_f_model = rf_grid_search.best_estimator_ # best model according to grid search \n",
    "\n",
    "best_random_f_model.get_params()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
