{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "-"
    }
   },
   "source": [
    "# Task Desription\n",
    "\n",
    "Please read first the following abstract\n",
    "\n",
    "## Abstract\n",
    "                \n",
    "RNA-Seq reveals an unprecedented complexity of the neuroblastoma transcriptome and is suitable for clinical endpoint prediction [ microarray ]\n",
    "\n",
    "### Experiment Description  \n",
    "\n",
    "We generated gene expression profiles from 498 primary neuroblastomas using RNA-Seq and microarrays. We sought to systematically evaluate the capability of RNA deep-sequencing (RNA-Seq)-based classification for clinical endpoint prediction in comparison to microarray-based ones. The neuroblastoma cohort was randomly divided into training and validation sets (**Please note:** <em>in the following we refer to this validation set as test set</em>), and 360 predictive models on six clinical endpoints were generated and evaluated. While prediction performances did not differ considerably between the two technical platforms, the RNA-Seq data processing pipelines, or feature levels (i.e., gene, transcript, and exon junction levels), RNA-Seq models based on the AceView database performed best on most endpoints. Collectively, our study reveals an unprecedented complexity of the neuroblastoma transcriptome, and provides guidelines for the development of gene expression-based predictive classifiers using high-throughput technologies.  Sample clinical characteristics definitions:  \n",
    "\n",
    "* sex: \n",
    "    <ul>\n",
    "    <li>M = male</li>\n",
    "    <li>F = female</li>\n",
    "    </ul>\n",
    "    \n",
    "* age at diagnosis: The age in days at diagnosis \n",
    "    <ul>\n",
    "    <li>integer</li>\n",
    "    </ul>\n",
    "\n",
    "* high risk: Clinically considered as high-risk neuroblastoma\n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "* INSS stage: Disease stage according to International Neuroblastoma Staging System ([INSS](https://www.cancer.org/cancer/neuroblastoma/detection-diagnosis-staging/staging.html)) \n",
    "    <ul>\n",
    "    <li>1</li>\n",
    "    <li>2</li>\n",
    "    <li>3</li>\n",
    "    <li>4</li>\n",
    "    <li>4S</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "* progression: Occurrence of a tumor progression event\n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "\n",
    "* death from disease: Occurrence of death from the disease (yes=1; no=0) \n",
    "    <ul>\n",
    "    <li>yes = 1</li>\n",
    "    <li>no = 0</li>\n",
    "    </ul>\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Gene expression of 498 neuroblastoma samples was quantified by RNA sequencing as well as by microarray analyses in order to understand the neuroblastoma transcriptome and predict clinical endpoints. \n",
    "\n",
    "\n",
    "## Task\n",
    "\n",
    "The task is to predict the missing values in the validation set (from here on called test set). Do this either with the RNASeq or the Microarray data, or potentially both?\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Code\n",
    "\n",
    "To make your start a bit easier, here is a small notebook reading the data in. It finishes with a function enabling you to save your predictions for submission. \n",
    "\n",
    "#### from here, the code starts\n",
    "\n",
    "First some imports "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt # plotting and visulisation\n",
    "import seaborn as sns # nicer (easier) visualisation\n",
    "%matplotlib inline\n",
    "\n",
    "\n",
    "# for saving\n",
    "import os,os.path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#data_dir ='/rds/homes/b/bjt186/MLP'.format(os.path.sep)\n",
    "\n",
    "#fn_fpkm             = 'log2FPKM_clean.tsv.gz'\n",
    "#fn_patient_info     = 'patientInfo.tsv'\n",
    "#fn_prop_intensities = 'allProbIntensities.tsv'\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the RNAs-Seq data\n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ID</th>\n",
       "      <th>NB001</th>\n",
       "      <th>NB002</th>\n",
       "      <th>NB003</th>\n",
       "      <th>NB004</th>\n",
       "      <th>NB005</th>\n",
       "      <th>NB006</th>\n",
       "      <th>NB007</th>\n",
       "      <th>NB008</th>\n",
       "      <th>NB009</th>\n",
       "      <th>NB010</th>\n",
       "      <th>...</th>\n",
       "      <th>NB489</th>\n",
       "      <th>NB490</th>\n",
       "      <th>NB491</th>\n",
       "      <th>NB492</th>\n",
       "      <th>NB493</th>\n",
       "      <th>NB494</th>\n",
       "      <th>NB495</th>\n",
       "      <th>NB496</th>\n",
       "      <th>NB497</th>\n",
       "      <th>NB498</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <td>0.834381</td>\n",
       "      <td>0.743094</td>\n",
       "      <td>0.909414</td>\n",
       "      <td>0.795775</td>\n",
       "      <td>0.905540</td>\n",
       "      <td>0.869154</td>\n",
       "      <td>1.811352</td>\n",
       "      <td>0.599240</td>\n",
       "      <td>0.981855</td>\n",
       "      <td>1.066399</td>\n",
       "      <td>...</td>\n",
       "      <td>0.997977</td>\n",
       "      <td>1.003559</td>\n",
       "      <td>0.842437</td>\n",
       "      <td>1.057873</td>\n",
       "      <td>0.805515</td>\n",
       "      <td>0.491331</td>\n",
       "      <td>0.868249</td>\n",
       "      <td>0.911379</td>\n",
       "      <td>0.660139</td>\n",
       "      <td>1.152988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.910053</td>\n",
       "      <td>0.941996</td>\n",
       "      <td>1.950857</td>\n",
       "      <td>1.989477</td>\n",
       "      <td>1.942946</td>\n",
       "      <td>1.927608</td>\n",
       "      <td>1.617745</td>\n",
       "      <td>2.161291</td>\n",
       "      <td>1.436439</td>\n",
       "      <td>2.159797</td>\n",
       "      <td>...</td>\n",
       "      <td>2.336929</td>\n",
       "      <td>2.836360</td>\n",
       "      <td>1.205317</td>\n",
       "      <td>2.439868</td>\n",
       "      <td>1.649027</td>\n",
       "      <td>1.451425</td>\n",
       "      <td>1.493852</td>\n",
       "      <td>1.641241</td>\n",
       "      <td>1.994978</td>\n",
       "      <td>1.289534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>1.453191</td>\n",
       "      <td>0.640614</td>\n",
       "      <td>1.156765</td>\n",
       "      <td>1.525277</td>\n",
       "      <td>1.365043</td>\n",
       "      <td>0.899212</td>\n",
       "      <td>1.304178</td>\n",
       "      <td>1.189205</td>\n",
       "      <td>0.771248</td>\n",
       "      <td>1.114787</td>\n",
       "      <td>...</td>\n",
       "      <td>1.182908</td>\n",
       "      <td>1.367371</td>\n",
       "      <td>0.643751</td>\n",
       "      <td>1.096815</td>\n",
       "      <td>0.925425</td>\n",
       "      <td>0.933275</td>\n",
       "      <td>1.208723</td>\n",
       "      <td>0.904511</td>\n",
       "      <td>1.529221</td>\n",
       "      <td>1.102866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0.005102</td>\n",
       "      <td>0.005902</td>\n",
       "      <td>0.005192</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.025347</td>\n",
       "      <td>0.005682</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021880</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024298</td>\n",
       "      <td>0.007295</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006678</td>\n",
       "      <td>0.005746</td>\n",
       "      <td>0.004998</td>\n",
       "      <td>0.004853</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.022780</td>\n",
       "      <td>0.018720</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0.580151</td>\n",
       "      <td>0.738233</td>\n",
       "      <td>0.927667</td>\n",
       "      <td>0.936497</td>\n",
       "      <td>0.924853</td>\n",
       "      <td>0.739038</td>\n",
       "      <td>1.018705</td>\n",
       "      <td>0.546324</td>\n",
       "      <td>0.666877</td>\n",
       "      <td>0.865850</td>\n",
       "      <td>...</td>\n",
       "      <td>0.673627</td>\n",
       "      <td>1.401265</td>\n",
       "      <td>0.837443</td>\n",
       "      <td>0.939849</td>\n",
       "      <td>0.743496</td>\n",
       "      <td>0.957837</td>\n",
       "      <td>0.812093</td>\n",
       "      <td>0.488748</td>\n",
       "      <td>1.068072</td>\n",
       "      <td>0.782887</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 498 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "ID              NB001     NB002     NB003     NB004     NB005     NB006  \\\n",
       "gene_id                                                                   \n",
       "1/2-SBSRNA4  0.834381  0.743094  0.909414  0.795775  0.905540  0.869154   \n",
       "A1BG         1.910053  0.941996  1.950857  1.989477  1.942946  1.927608   \n",
       "A1BG-AS1     1.453191  0.640614  1.156765  1.525277  1.365043  0.899212   \n",
       "A1CF         0.005102  0.005902  0.005192  0.000000  0.025347  0.005682   \n",
       "A2LD1        0.580151  0.738233  0.927667  0.936497  0.924853  0.739038   \n",
       "\n",
       "ID              NB007     NB008     NB009     NB010  ...     NB489     NB490  \\\n",
       "gene_id                                              ...                       \n",
       "1/2-SBSRNA4  1.811352  0.599240  0.981855  1.066399  ...  0.997977  1.003559   \n",
       "A1BG         1.617745  2.161291  1.436439  2.159797  ...  2.336929  2.836360   \n",
       "A1BG-AS1     1.304178  1.189205  0.771248  1.114787  ...  1.182908  1.367371   \n",
       "A1CF         0.000000  0.000000  0.021880  0.000000  ...  0.024298  0.007295   \n",
       "A2LD1        1.018705  0.546324  0.666877  0.865850  ...  0.673627  1.401265   \n",
       "\n",
       "ID              NB491     NB492     NB493     NB494     NB495     NB496  \\\n",
       "gene_id                                                                   \n",
       "1/2-SBSRNA4  0.842437  1.057873  0.805515  0.491331  0.868249  0.911379   \n",
       "A1BG         1.205317  2.439868  1.649027  1.451425  1.493852  1.641241   \n",
       "A1BG-AS1     0.643751  1.096815  0.925425  0.933275  1.208723  0.904511   \n",
       "A1CF         0.000000  0.006678  0.005746  0.004998  0.004853  0.000000   \n",
       "A2LD1        0.837443  0.939849  0.743496  0.957837  0.812093  0.488748   \n",
       "\n",
       "ID              NB497     NB498  \n",
       "gene_id                          \n",
       "1/2-SBSRNA4  0.660139  1.152988  \n",
       "A1BG         1.994978  1.289534  \n",
       "A1BG-AS1     1.529221  1.102866  \n",
       "A1CF         0.022780  0.018720  \n",
       "A2LD1        1.068072  0.782887  \n",
       "\n",
       "[5 rows x 498 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_fpkm = pd.read_csv('{}/{}'.format(data_dir,fn_fpkm),sep='\\t',compression='.gz').rename({'00gene_id':'gene_id'},axis=1)\n",
    "\n",
    "df_fpkm = pd.read_csv('log2FPKM.tsv.gz',sep='\\t').rename({'00gene_id':'gene_id'},axis=1)\n",
    "\n",
    "#df = pd.read_csv('filename.zip', compression='zip', header=0, sep=',', quotechar='\"')\n",
    "\n",
    "df_fpkm = df_fpkm.set_index(['gene_id'])\n",
    "df_fpkm.columns.name = 'ID'\n",
    "\n",
    "df_fpkm.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the Microarray data\n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#df_prop_intensities = pd.read_csv('{}/{}'.format(data_dir,fn_prop_intensities),sep='\\t').set_index(['Reporter.Identifier'])\n",
    "#df_prop_intensities.columns.name = 'ID'\n",
    "\n",
    "#df_prop_intensities.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Load the patient factors, including the potential endpoints \n",
    "\n",
    "This part already sets the indeces in the DataFrame. Please feel free to change as required. \n",
    "Please note, that the ```FactorValues``` should have a 1-to-1 correspondence to the factors desc ribed in the abstract. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>ID</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NB498</td>\n",
       "      <td>female</td>\n",
       "      <td>530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NB497</td>\n",
       "      <td>female</td>\n",
       "      <td>379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NB496</td>\n",
       "      <td>male</td>\n",
       "      <td>132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NB495</td>\n",
       "      <td>male</td>\n",
       "      <td>163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NB494</td>\n",
       "      <td>male</td>\n",
       "      <td>56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues     ID FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "0             NB498            female                             530   \n",
       "1             NB497            female                             379   \n",
       "2             NB496              male                             132   \n",
       "3             NB495              male                             163   \n",
       "4             NB494              male                              56   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "0                                          NaN                      NaN   \n",
       "1                                          0.0                      0.0   \n",
       "2                                          NaN                      NaN   \n",
       "3                                          0.0                      0.0   \n",
       "4                                          NaN                      NaN   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "0                                 NaN                        NaN  \n",
       "1                                   1                        0.0  \n",
       "2                                 NaN                        NaN  \n",
       "3                                   1                        0.0  \n",
       "4                                 NaN                        NaN  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#df_patient_info = pd.read_csv('{}/{}'.format(data_dir,fn_patient_info),sep='\\t').set_index('ID')\n",
    "df_patient_info = pd.read_csv('patientInfo.tsv',sep='\\t')\n",
    "df_patient_info.columns.name = 'FactorValues'\n",
    "\n",
    "df_patient_info.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####  Divide into training and external testing\n",
    "\n",
    "As you might have already noticed, we removed some of the factor values for some of the patient **ID**s.\n",
    "Every row, where this information is missing indicate a real validation entry. We can use this information and create two separate DataFrames, one for training, one for the validation (testing). \n",
    "\n",
    "The task is to predict the missing values, either with the RNASeq or the Microarray data, or potentially both?\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_patient_info_train  = df_patient_info[df_patient_info['FactorValue..death.from.disease.'].notna()]\n",
    "df_patient_info_test   = df_patient_info[df_patient_info['FactorValue..death.from.disease.'].isna()]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>ID</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NB497</td>\n",
       "      <td>female</td>\n",
       "      <td>379</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NB495</td>\n",
       "      <td>male</td>\n",
       "      <td>163</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>NB493</td>\n",
       "      <td>male</td>\n",
       "      <td>190</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>NB491</td>\n",
       "      <td>male</td>\n",
       "      <td>2326</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NB489</td>\n",
       "      <td>female</td>\n",
       "      <td>865</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues     ID FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "1             NB497            female                             379   \n",
       "3             NB495              male                             163   \n",
       "5             NB493              male                             190   \n",
       "7             NB491              male                            2326   \n",
       "9             NB489            female                             865   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "1                                          0.0                      0.0   \n",
       "3                                          0.0                      0.0   \n",
       "5                                          0.0                      0.0   \n",
       "7                                          0.0                      1.0   \n",
       "9                                          0.0                      1.0   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "1                                   1                        0.0  \n",
       "3                                   1                        0.0  \n",
       "5                                   1                        0.0  \n",
       "7                                   4                        1.0  \n",
       "9                                   4                        0.0  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info_train.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>FactorValues</th>\n",
       "      <th>ID</th>\n",
       "      <th>FactorValue..Sex.</th>\n",
       "      <th>FactorValue..age.at.diagnosis.</th>\n",
       "      <th>FactorValue..death.from.disease.</th>\n",
       "      <th>FactorValue..high.risk.</th>\n",
       "      <th>FactorValue..inss.stage.</th>\n",
       "      <th>FactorValue..progression.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NB498</td>\n",
       "      <td>female</td>\n",
       "      <td>530</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NB496</td>\n",
       "      <td>male</td>\n",
       "      <td>132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NB494</td>\n",
       "      <td>male</td>\n",
       "      <td>56</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>NB492</td>\n",
       "      <td>male</td>\n",
       "      <td>947</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>NB490</td>\n",
       "      <td>female</td>\n",
       "      <td>1759</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "FactorValues     ID FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
       "0             NB498            female                             530   \n",
       "2             NB496              male                             132   \n",
       "4             NB494              male                              56   \n",
       "6             NB492              male                             947   \n",
       "8             NB490            female                            1759   \n",
       "\n",
       "FactorValues  FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
       "0                                          NaN                      NaN   \n",
       "2                                          NaN                      NaN   \n",
       "4                                          NaN                      NaN   \n",
       "6                                          NaN                      NaN   \n",
       "8                                          NaN                      NaN   \n",
       "\n",
       "FactorValues FactorValue..inss.stage.  FactorValue..progression.  \n",
       "0                                 NaN                        NaN  \n",
       "2                                 NaN                        NaN  \n",
       "4                                 NaN                        NaN  \n",
       "6                                 NaN                        NaN  \n",
       "8                                 NaN                        NaN  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_patient_info_test.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['NB001', 'NB002', 'NB003', 'NB004', 'NB005', 'NB006', 'NB007', 'NB008', 'NB009', 'NB010', 'NB011', 'NB012', 'NB013', 'NB014', 'NB015', 'NB016', 'NB017', 'NB018', 'NB019', 'NB020', 'NB021', 'NB022', 'NB023', 'NB024', 'NB025', 'NB026', 'NB027', 'NB028', 'NB029', 'NB030', 'NB031', 'NB032', 'NB033', 'NB034', 'NB035', 'NB036', 'NB037', 'NB038', 'NB039', 'NB040', 'NB041', 'NB042', 'NB043', 'NB044', 'NB045', 'NB046', 'NB047', 'NB048', 'NB049', 'NB050', 'NB051', 'NB052', 'NB053', 'NB054', 'NB055', 'NB056', 'NB057', 'NB058', 'NB059', 'NB060', 'NB061', 'NB062', 'NB063', 'NB064', 'NB065', 'NB066', 'NB067', 'NB068', 'NB069', 'NB070', 'NB071', 'NB072', 'NB073', 'NB074', 'NB075', 'NB076', 'NB077', 'NB078', 'NB079', 'NB080', 'NB081', 'NB082', 'NB083', 'NB084', 'NB085', 'NB086', 'NB087', 'NB088', 'NB089', 'NB090', 'NB091', 'NB092', 'NB093', 'NB094', 'NB095', 'NB096', 'NB097', 'NB098', 'NB099', 'NB100', 'NB101', 'NB102', 'NB103', 'NB104', 'NB105', 'NB106', 'NB107', 'NB108', 'NB109', 'NB110', 'NB111', 'NB112', 'NB113', 'NB114', 'NB115', 'NB116', 'NB117', 'NB118', 'NB119', 'NB120', 'NB121', 'NB122', 'NB123', 'NB124', 'NB125', 'NB126', 'NB127', 'NB128', 'NB129', 'NB130', 'NB131', 'NB132', 'NB133', 'NB134', 'NB135', 'NB136', 'NB137', 'NB138', 'NB139', 'NB140', 'NB141', 'NB142', 'NB143', 'NB144', 'NB145', 'NB146', 'NB147', 'NB148', 'NB149', 'NB150', 'NB151', 'NB152', 'NB153', 'NB154', 'NB155', 'NB156', 'NB157', 'NB158', 'NB159', 'NB160', 'NB161', 'NB162', 'NB163', 'NB164', 'NB165', 'NB166', 'NB167', 'NB168', 'NB169', 'NB170', 'NB171', 'NB172', 'NB173', 'NB174', 'NB175', 'NB176', 'NB177', 'NB178', 'NB179', 'NB180', 'NB181', 'NB182', 'NB183', 'NB184', 'NB185', 'NB186', 'NB187', 'NB188', 'NB189', 'NB190', 'NB191', 'NB192', 'NB193', 'NB194', 'NB195', 'NB196', 'NB197', 'NB198', 'NB199', 'NB200', 'NB201', 'NB202', 'NB203', 'NB204', 'NB205', 'NB206', 'NB207', 'NB208', 'NB209', 'NB210', 'NB211', 'NB212', 'NB213', 'NB214', 'NB215', 'NB216', 'NB217', 'NB218', 'NB219', 'NB220', 'NB221', 'NB222', 'NB223', 'NB224', 'NB225', 'NB226', 'NB227', 'NB228', 'NB229', 'NB230', 'NB231', 'NB232', 'NB233', 'NB234', 'NB235', 'NB236', 'NB237', 'NB238', 'NB239', 'NB240', 'NB241', 'NB242', 'NB243', 'NB244', 'NB245', 'NB246', 'NB247', 'NB248', 'NB249', 'NB250', 'NB251', 'NB252', 'NB253', 'NB254', 'NB255', 'NB256', 'NB257', 'NB258', 'NB259', 'NB260', 'NB261', 'NB262', 'NB263', 'NB264', 'NB265', 'NB266', 'NB267', 'NB268', 'NB269', 'NB270', 'NB271', 'NB272', 'NB273', 'NB274', 'NB275', 'NB276', 'NB277', 'NB278', 'NB279', 'NB280', 'NB281', 'NB282', 'NB283', 'NB284', 'NB285', 'NB286', 'NB287', 'NB288', 'NB289', 'NB290', 'NB291', 'NB292', 'NB293', 'NB294', 'NB295', 'NB296', 'NB297', 'NB298', 'NB299', 'NB300', 'NB301', 'NB302', 'NB303', 'NB304', 'NB305', 'NB306', 'NB307', 'NB308', 'NB309', 'NB310', 'NB311', 'NB312', 'NB313', 'NB314', 'NB315', 'NB316', 'NB317', 'NB318', 'NB319', 'NB320', 'NB321', 'NB322', 'NB323', 'NB324', 'NB325', 'NB326', 'NB327', 'NB328', 'NB329', 'NB330', 'NB331', 'NB332', 'NB333', 'NB334', 'NB335', 'NB336', 'NB337', 'NB338', 'NB339', 'NB340', 'NB341', 'NB342', 'NB343', 'NB344', 'NB345', 'NB346', 'NB347', 'NB348', 'NB349', 'NB350', 'NB351', 'NB352', 'NB353', 'NB354', 'NB355', 'NB356', 'NB357', 'NB358', 'NB359', 'NB360', 'NB361', 'NB362', 'NB363', 'NB364', 'NB365', 'NB366', 'NB367', 'NB368', 'NB369', 'NB370', 'NB371', 'NB372', 'NB373', 'NB374', 'NB375', 'NB376', 'NB377', 'NB378', 'NB379', 'NB380', 'NB381', 'NB382', 'NB383', 'NB384', 'NB385', 'NB386', 'NB387', 'NB388', 'NB389', 'NB390', 'NB391', 'NB392', 'NB393', 'NB394', 'NB395', 'NB396', 'NB397', 'NB398', 'NB399', 'NB400', 'NB401', 'NB402', 'NB403', 'NB404', 'NB405', 'NB406', 'NB407', 'NB408', 'NB409', 'NB410', 'NB411', 'NB412', 'NB413', 'NB414', 'NB415', 'NB416', 'NB417', 'NB418', 'NB419', 'NB420', 'NB421', 'NB422', 'NB423', 'NB424', 'NB425', 'NB426', 'NB427', 'NB428', 'NB429', 'NB430', 'NB431', 'NB432', 'NB433', 'NB434', 'NB435', 'NB436', 'NB437', 'NB438', 'NB439', 'NB440', 'NB441', 'NB442', 'NB443', 'NB444', 'NB445', 'NB446', 'NB447', 'NB448', 'NB449', 'NB450', 'NB451', 'NB452', 'NB453', 'NB454', 'NB455', 'NB456', 'NB457', 'NB458', 'NB459', 'NB460', 'NB461', 'NB462', 'NB463', 'NB464', 'NB465', 'NB466', 'NB467', 'NB468', 'NB469', 'NB470', 'NB471', 'NB472', 'NB473', 'NB474', 'NB475', 'NB476', 'NB477', 'NB478', 'NB479', 'NB480', 'NB481', 'NB482', 'NB483', 'NB484', 'NB485', 'NB486', 'NB487', 'NB488', 'NB489', 'NB490', 'NB491', 'NB492', 'NB493', 'NB494', 'NB495', 'NB496', 'NB497', 'NB498']\n"
     ]
    }
   ],
   "source": [
    "x_posed_SEQ = df_fpkm.T\n",
    "#print(x_posed_SEQ.iloc[150:200,:])\n",
    "# load the index as a column called Idents\n",
    "x_posed_SEQ['Idents'] = x_posed_SEQ.index\n",
    "\n",
    "# create Idents as list\n",
    "Idents_list = list(x_posed_SEQ['Idents'])\n",
    "#print(Idents_list[0:100])\n",
    "\n",
    "#str = \"NB013_BC02FPACXX_L1_CGATGT\"\n",
    "#print(str[0:5])\n",
    "\n",
    "#create empty list to hold clean idents\n",
    "trimmed_idents = []\n",
    "\n",
    "# iterate\n",
    "for ident in Idents_list:\n",
    "    new_ident = ident[0:5]\n",
    "    trimmed_idents.append(new_ident)\n",
    "\n",
    "\n",
    "#load clean idents back into dataframe as index\n",
    "x_posed_SEQ.index = trimmed_idents\n",
    "# and as an ID column to allow for the merge\n",
    "x_posed_SEQ['ID'] = trimmed_idents\n",
    "# drop the Idents column\n",
    "clean_SEQ = x_posed_SEQ.drop('Idents', 1)\n",
    "\n",
    "print(clean_SEQ.shape)\n",
    "print(clean_SEQ.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<bound method NDFrame.head of      1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M     A2ML1  \\\n",
      "0       0.834381  1.910053  1.453191  0.005102  0.580151  5.137167  0.027558   \n",
      "1       0.909414  1.950857  1.156765  0.005192  0.927667  5.405414  0.028036   \n",
      "2       0.905540  1.942946  1.365043  0.025347  0.924853  5.613060  0.000000   \n",
      "3       1.811352  1.617745  1.304178  0.000000  1.018705  6.956301  0.000000   \n",
      "4       0.981855  1.436439  0.771248  0.021880  0.666877  5.170205  0.000000   \n",
      "..           ...       ...       ...       ...       ...       ...       ...   \n",
      "244     0.997977  2.336929  1.182908  0.024298  0.673627  5.266052  0.000000   \n",
      "245     0.842437  1.205317  0.643751  0.000000  0.837443  5.049343  0.000000   \n",
      "246     0.805515  1.649027  0.925425  0.005746  0.743496  5.223052  0.000000   \n",
      "247     0.868249  1.493852  1.208723  0.004853  0.812093  6.655558  0.000000   \n",
      "248     0.660139  1.994978  1.529221  0.022780  1.068072  6.581892  0.000000   \n",
      "\n",
      "        A2MP1    A4GALT     A4GNT  ...       ZYX     ZZEF1      ZZZ3     ID  \\\n",
      "0    0.170671  1.535670  0.057101  ...  5.039327  3.245179  3.296496  NB001   \n",
      "1    0.045284  0.943194  0.000000  ...  4.982975  2.880531  2.431521  NB003   \n",
      "2    0.000000  1.549078  0.032056  ...  5.236531  2.761698  2.226502  NB005   \n",
      "3    0.210127  1.959344  0.164813  ...  5.645258  2.755304  2.933508  NB007   \n",
      "4    0.000000  1.374772  0.031028  ...  5.393207  2.730553  2.271136  NB009   \n",
      "..        ...       ...       ...  ...       ...       ...       ...    ...   \n",
      "244  0.053645  0.854657  0.034626  ...  5.103993  3.519328  2.563558  NB489   \n",
      "245  0.000000  1.025780  0.036918  ...  4.938587  2.949716  2.957478  NB491   \n",
      "246  0.050377  0.830303  0.000000  ...  4.877509  2.951646  3.293082  NB493   \n",
      "247  0.124005  1.585030  0.157478  ...  5.207842  3.231953  3.206993  NB495   \n",
      "248  0.000000  1.544703  0.057179  ...  5.427338  3.107112  3.446606  NB497   \n",
      "\n",
      "     FactorValue..Sex.  FactorValue..age.at.diagnosis.  \\\n",
      "0                 male                             987   \n",
      "1               female                             625   \n",
      "2               female                             536   \n",
      "3                 male                             942   \n",
      "4               female                             217   \n",
      "..                 ...                             ...   \n",
      "244             female                             865   \n",
      "245               male                            2326   \n",
      "246               male                             190   \n",
      "247               male                             163   \n",
      "248             female                             379   \n",
      "\n",
      "     FactorValue..death.from.disease.  FactorValue..high.risk.  \\\n",
      "0                                 1.0                      1.0   \n",
      "1                                 1.0                      1.0   \n",
      "2                                 1.0                      1.0   \n",
      "3                                 0.0                      1.0   \n",
      "4                                 0.0                      1.0   \n",
      "..                                ...                      ...   \n",
      "244                               0.0                      1.0   \n",
      "245                               0.0                      1.0   \n",
      "246                               0.0                      0.0   \n",
      "247                               0.0                      0.0   \n",
      "248                               0.0                      0.0   \n",
      "\n",
      "     FactorValue..inss.stage.  FactorValue..progression.  \n",
      "0                           4                        1.0  \n",
      "1                           4                        1.0  \n",
      "2                           4                        1.0  \n",
      "3                           4                        1.0  \n",
      "4                          4S                        0.0  \n",
      "..                        ...                        ...  \n",
      "244                         4                        0.0  \n",
      "245                         4                        1.0  \n",
      "246                         1                        0.0  \n",
      "247                         1                        0.0  \n",
      "248                         1                        0.0  \n",
      "\n",
      "[249 rows x 23153 columns]>\n",
      "23146\n",
      "(249, 23146)\n",
      "   1/2-SBSRNA4      A1BG  A1BG-AS1      A1CF     A2LD1       A2M     A2ML1  \\\n",
      "0     0.834381  1.910053  1.453191  0.005102  0.580151  5.137167  0.027558   \n",
      "1     0.909414  1.950857  1.156765  0.005192  0.927667  5.405414  0.028036   \n",
      "2     0.905540  1.942946  1.365043  0.025347  0.924853  5.613060  0.000000   \n",
      "3     1.811352  1.617745  1.304178  0.000000  1.018705  6.956301  0.000000   \n",
      "4     0.981855  1.436439  0.771248  0.021880  0.666877  5.170205  0.000000   \n",
      "\n",
      "      A2MP1    A4GALT     A4GNT  ...    ZWILCH     ZWINT      ZXDA      ZXDB  \\\n",
      "0  0.170671  1.535670  0.057101  ...  3.651545  5.045111  1.575337  2.395551   \n",
      "1  0.045284  0.943194  0.000000  ...  3.547080  4.721531  1.508530  2.482464   \n",
      "2  0.000000  1.549078  0.032056  ...  3.582977  4.766362  1.038738  1.890104   \n",
      "3  0.210127  1.959344  0.164813  ...  3.475448  4.009473  1.407451  2.198063   \n",
      "4  0.000000  1.374772  0.031028  ...  3.581122  5.193176  1.196494  2.289598   \n",
      "\n",
      "       ZXDC    ZYG11A    ZYG11B       ZYX     ZZEF1      ZZZ3  \n",
      "0  3.466692  0.862512  3.561729  5.039327  3.245179  3.296496  \n",
      "1  3.056684  0.022912  2.236695  4.982975  2.880531  2.431521  \n",
      "2  2.823576  0.049621  2.235632  5.236531  2.761698  2.226502  \n",
      "3  3.123347  0.044684  3.130989  5.645258  2.755304  2.933508  \n",
      "4  3.409242  0.012183  2.361504  5.393207  2.730553  2.271136  \n",
      "\n",
      "[5 rows x 23146 columns]\n"
     ]
    }
   ],
   "source": [
    "Merged_Data_patient_train=clean_SEQ.merge(df_patient_info_train, on='ID', how='inner')\n",
    "clean_patient_train = Merged_Data_patient_train.iloc[:,:-7]\n",
    "\n",
    "print((Merged_Data_patient_train).head)\n",
    "\n",
    "lists = clean_patient_train.columns.values.tolist()\n",
    "\n",
    "print(len(lists))\n",
    "print(clean_patient_train.shape)\n",
    "#print(clean_patient_train.head())\n",
    "# ok that looks good so we can drop the id column now\n",
    "#clean_patient_train = clean_patient_train.drop('ID', 1)\n",
    "print(clean_patient_train.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis\n",
    "\n",
    "From here on, you will need to use your skills ...\n",
    "\n",
    "### Note:\n",
    "To be clear, there are **multiple** target features/attributes to predict. Say you want to build a model predicting **death from disease** of a patient, your target variable is ```'FactorValue..death.from.disease.'``` and the corresponding target vector <em>y</em> would be as follows:\n",
    "\n",
    "```python\n",
    "y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "```\n",
    "Taking the other data into account (RNASeq or microarray) as ``` X_train``` (you will have to preprocess and split this information yourselves), you could for example build a random forest model:\n",
    "\n",
    "```python\n",
    "from sklearn.ensemble import RandomForestClassifier\n",
    "random_f_model_death = random_f_model = RandomForestClassifier() \n",
    "random_f_model_death.fit(X_train,y_train)\n",
    "```\n",
    "and predict ```y_test``` using ```X_test```. \n",
    "\n",
    "Obviously, you want to avoid any overfitting and might want to use appropriate validation approaches. \n",
    "\n",
    "Once you have your model and the prediction for the test data, you should be able to fill the ```'FactorValue..death.from.disease.'``` column in the test set. \n",
    "\n",
    "For the submission, please also include the confidence/probability/score for each of the prediction (assume ```1``` to be the value for the positive class). This only applies to the factors:  **high risk**, **progression** and **death from disease** . \n",
    "\n",
    "The other two factors need to be treated differently.  \n",
    "\n",
    "Using this very rough outline, you should be able to predict all factors in the test data. \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNASeq\n",
      "(23146, 498)\n",
      "Patient info train\n",
      "(249, 7)\n",
      "Patient info test\n",
      "(249, 7)\n"
     ]
    }
   ],
   "source": [
    "print(\"RNASeq\")\n",
    "\n",
    "print(np.shape(df_fpkm))\n",
    "\n",
    "# size of the patient info training set\n",
    "\n",
    "print(\"Patient info train\")\n",
    "\n",
    "print(np.shape(df_patient_info_train))\n",
    "\n",
    "# and the size of the patient info test set\n",
    "\n",
    "print(\"Patient info test\")\n",
    "\n",
    "print(np.shape(df_patient_info_test))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ID', 'FactorValue..Sex.', 'FactorValue..age.at.diagnosis.', 'FactorValue..death.from.disease.', 'FactorValue..high.risk.', 'FactorValue..inss.stage.', 'FactorValue..progression.']\n",
      "['1', '1', '1', '4', '4', '4', '3', '1', '4', '1']\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXAAAAD4CAYAAAD1jb0+AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8/fFQqAAAACXBIWXMAAAsTAAALEwEAmpwYAAALhUlEQVR4nO3dYaidh13H8e/PpqWbm7ZbL6UmjTewUgmDtnKplYBIoxCXuhYsozJLlEgUNs3cpMt8M/YuA1m7FyKEpRqw1I220LKCUroUESQz6TK7NhsLNZkp6ZK51U1fbEb/vriPa5ae9J57zz05959+PxDueZ7zPPf589Dz5elz7rk3VYUkqZ+fmvUAkqSVMeCS1JQBl6SmDLgkNWXAJampdZfyYNddd13Nz89fykNKUntHjhz5TlXNXbj+kgZ8fn6ew4cPX8pDSlJ7SU6OWu8tFElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrqkn4SU8szv+fpmRz3xN7tMzmupOXxClySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqamxAp7kT5K8mORrSR5NcnWSTUkOJTme5PNJrpr2sJKk1y0Z8CTrgT8GFqrqvcAVwH3Ap4EHq+o9wPeAndMcVJL0k8a9hbIOeFuSdcDbgdPAncBjw/MHgHtWfTpJ0kUtGfCqegX4c+BbLIb7P4AjwGtVdW7Y7BSwftT+SXYlOZzk8NmzZ1dnaknSWLdQrgXuBjYBPwf8NLBt3ANU1b6qWqiqhbm5uRUPKkn6SePcQvk14F+r6mxV/TfwBLAFuGa4pQKwAXhlSjNKkkYYJ+DfAu5I8vYkAbYCLwEHgXuHbXYAT05nREnSKOPcAz/E4puVzwMvDPvsAz4OfDTJceDdwP4pzilJusC6pTeBqvok8MkLVr8M3L7qE0mSxuInMSWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSU2MFPMk1SR5L8vUkx5L8cpJ3JXkmyTeHr9dOe1hJ0uvGvQL/LPB3VfULwC3AMWAP8GxV3QQ8OyxLki6RJQOe5GeBXwH2A1TVj6rqNeBu4MCw2QHgnumMKEkaZd0Y22wCzgJ/leQW4AiwG7i+qk4P27wKXD9q5yS7gF0AGzduXPGg83ueXvG+kzixd/tMjitJSxnnFso64BeBv6yq24D/4oLbJVVVQI3auar2VdVCVS3Mzc1NOq8kaTBOwE8Bp6rq0LD8GItB/3aSGwCGr2emM6IkaZQlA15VrwL/luTmYdVW4CXgKWDHsG4H8ORUJpQkjTTOPXCAPwIeSXIV8DLweyzG/wtJdgIngQ9MZ0RJ0ihjBbyqjgILI57auqrTSJLG5icxJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmho74EmuSPKVJF8cljclOZTkeJLPJ7lqemNKki60nCvw3cCx85Y/DTxYVe8BvgfsXM3BJElvbqyAJ9kAbAc+NywHuBN4bNjkAHDPFOaTJF3EujG3ewh4AHjnsPxu4LWqOjcsnwLWj9oxyS5gF8DGjRtXPKik1TW/5+mZHfvE3u0zO/blZMkr8CR3AWeq6shKDlBV+6pqoaoW5ubmVvItJEkjjHMFvgV4f5L3AVcDPwN8FrgmybrhKnwD8Mr0xpQkXWjJK/Cq+kRVbaiqeeA+4EtV9UHgIHDvsNkO4MmpTSlJeoNJfg7848BHkxxn8Z74/tUZSZI0jnHfxASgqp4DnhsevwzcvvojSZLG4ScxJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpqWX9RR7pcjW/5+mZHfvE3u0zO7Z68wpckpoy4JLUlAGXpKa8By7pLWNW73VM630Or8AlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1NSSAU9yY5KDSV5K8mKS3cP6dyV5Jsk3h6/XTn9cSdL/G+cK/BzwsaraDNwBfCjJZmAP8GxV3QQ8OyxLki6RJQNeVaer6vnh8Q+AY8B64G7gwLDZAeCeKc0oSRphWffAk8wDtwGHgOur6vTw1KvA9RfZZ1eSw0kOnz17dpJZJUnnGTvgSd4BPA58pKq+f/5zVVVAjdqvqvZV1UJVLczNzU00rCTpdWMFPMmVLMb7kap6Ylj97SQ3DM/fAJyZzoiSpFHG+SmUAPuBY1X1mfOeegrYMTzeATy5+uNJki5mnL9KvwW4H3ghydFh3Z8Be4EvJNkJnAQ+MJUJJUkjLRnwqvpHIBd5euvqjiNJGpefxJSkpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWrKgEtSUwZckpoy4JLUlAGXpKYMuCQ1ZcAlqSkDLklNGXBJasqAS1JTBlySmjLgktSUAZekpgy4JDVlwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1JQBl6SmDLgkNWXAJakpAy5JTRlwSWpqooAn2ZbkG0mOJ9mzWkNJkpa24oAnuQL4C+A3gM3AbyfZvFqDSZLe3CRX4LcDx6vq5ar6EfC3wN2rM5YkaSmpqpXtmNwLbKuq3x+W7wd+qao+fMF2u4Bdw+LNwDdWOOt1wHdWuO9bkedreTxfy+P5Wp5Jz9fPV9XchSvXTfANx1JV+4B9k36fJIeramEVRnpL8Hwtj+dreTxfyzOt8zXJLZRXgBvPW94wrJMkXQKTBPyfgZuSbEpyFXAf8NTqjCVJWsqKb6FU1bkkHwb+HrgCeLiqXly1yd5o4tswbzGer+XxfC2P52t5pnK+VvwmpiRptvwkpiQ1ZcAlqak1H/AkDyc5k+Rrs56liyRXJPlKki/Oepa1LsnVSb6c5KtJXkzyqVnPtJYluTHJwSQvDedr96xnWqsufB0muWtY/upw/v5g0mOs+YADfw1sm/UQzewGjs16iCZ+CNxZVbcAtwLbktwx25HWtHPAx6pqM3AH8CF/hcZF/fh1mORKFt/I/M3hv7XbgOcmPcCaD3hV/QPw3VnP0UWSDcB24HOznqWDWvSfw+KVwz/f2b+IqjpdVc8Pj3/AYqDWz3aqtWfE6/CdLP7U378DVNUPq2qln0r/sTUfcC3bQ8ADwP/OeI42hv/VPQqcAZ6pqkMzHqmFJPMsXkl6vt7oIc57HVbVd1n8nMzJJI8m+WCSiftrwC8jSe4CzlTVkVnP0klV/U9V3crip4lvT/LeGY+05iV5B/A48JGq+v6s51lLLvY6HH5v1Fbgy8CfAg9PeiwDfnnZArw/yQkWfzvknUn+ZrYj9VFVrwEH8T2XNzXcz30ceKSqnpj1PGvQRV+HVfVCVT0I/DrwW5MeyIBfRqrqE1W1oarmWfzVBl+qqt+Z8VhrWpK5JNcMj9/G4gvr6zMdag1LEmA/cKyqPjPredaiUa9D4A+T/Op5m90KnJz0WGs+4EkeBf4JuDnJqSQ7Zz2TLis3AAeT/AuLv9/nmaryxy8vbgtwP4tXlUeHf++b9VANBHhg+AtmR4FPAb878Tf1o/SS1NOavwKXJI1mwCWpKQMuSU0ZcElqyoBLUlMGXJKaMuCS1NT/Aba3a9uLSbNHAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "# have a look at the INSS classifications\n",
    "\n",
    "headers_list = list(df_patient_info_train.columns)\n",
    "print(headers_list)\n",
    "INSS_list = list(df_patient_info_train.iloc[:,5]) #INSS Stage\n",
    "print(INSS_list[0:10])\n",
    "\n",
    "from collections import Counter\n",
    "INSS_cats = Counter(INSS_list)\n",
    "plt.hist(INSS_list)\n",
    "INSS_dict = {\"1\":60, \"2\":40, \"3\":30, \"4\":91, \"4S\":28}\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "RandomForestClassifier()"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "# using the RNA data as the X train\n",
    "\n",
    "X_train = df_fpkm[0:249]\n",
    "\n",
    "#looking for the people dying from the disease as the training set\n",
    "y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "\n",
    "random_f_model_death = random_f_model = RandomForestClassifier()\n",
    "random_f_model_death.fit(X_train,y_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(x_posed_SEQ.head)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(df_patient_info_train.iloc[150:200,:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#print(Merged_Data_patient_train)\n",
    "first_column = Merged_Data_patient_train.iloc[:, 0]\n",
    "print(first_column)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "type(Merged_Data_patient_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.ensemble import RandomForestClassifier\n",
    "\n",
    "X_train = clean_patient_train\n",
    "\n",
    "# looking for the people dying from the disease as the training set\n",
    "y_train = df_patient_info_train['FactorValue..death.from.disease.'].astype(int)\n",
    "\n",
    "random_f_model_death = random_f_model = RandomForestClassifier()\n",
    "random_f_model_death.fit(X_train,y_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import train_test_split function\n",
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "X=clean_patient_train  # Features\n",
    "y=df_patient_info_train['FactorValue..death.from.disease.'].astype(int)  # Labels\n",
    "\n",
    "# Split dataset into training set and test set\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "clf=RandomForestClassifier(n_estimators=100)\n",
    "\n",
    "#Train the model using the training sets y_pred=clf.predict(X_test)\n",
    "clf.fit(X_train,y_train)\n",
    "\n",
    "# predictions\n",
    "y_pred=clf.predict(X_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import scikit-learn metrics module for accuracy calculation\n",
    "from sklearn import metrics\n",
    "# Model Accuracy, how often is the classifier correct?\n",
    "print(\"Accuracy:\",metrics.accuracy_score(y_test, y_pred))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "feature_imp = pd.Series(clf.feature_importances_,index=lists).sort_values(ascending=False)\n",
    "feature_imp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "# Creating a bar plot\n",
    "sns.barplot(x=feature_imp, y=lists)\n",
    "# Add labels to your graph\n",
    "plt.xlabel('Feature Importance Score')\n",
    "plt.ylabel('Features')\n",
    "plt.title(\"Visualizing Important Features\")\n",
    "plt.legend()\n",
    "plt.ylim(0, 10)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_patient_info_train['FactorValue..Sex._male'].replace({'male':1, 'female':0}, inplace=True)\n",
    "print(df_patient_info_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_patient_info_train.rename(columns={'FactorValue..inss.stage._4S' : 'FactorValue..inss.stage._5'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.ensemble import RandomForestRegressor\n",
    "\n",
    "independentVariables = clean_patient_test\n",
    "rfModel_death = RandomForestRegressor()\n",
    "rfModel_death.fit(independentVariables, df_patient_info_train['FactorValue..death.from.disease.'].astype(int))\n",
    "\n",
    "generatedDeathValues = rfModel_death.predict(df_patient_info_test)\n",
    "print(generatedDeathValues)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
